<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Paul Hook" />


<title>cds_creation-qc_filtering</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/simplex.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 41px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 46px;
  margin-top: -46px;
}

.section h2 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h3 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h4 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h5 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h6 {
  padding-top: 46px;
  margin-top: -46px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">sc-da-parkinsons</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="cds_creation-qc_filtering.html">CDS creation and QC</a>
</li>
<li>
  <a href="E15.clustering.html">E15.5 Analyses</a>
</li>
<li>
  <a href="P7.analyses.links.html">P7 Analyses</a>
</li>
<li>
  <a href="All.cells.analyses.links.html">All cell analyses</a>
</li>
<li>
  <a href="PD.GWAS.scoring.links.html">PD Gene Scoring</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/pwh124/sc-da-parkinsons">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">cds_creation-qc_filtering</h1>
<h4 class="author"><em>Paul Hook</em></h4>

</div>


<p><strong>Last update:</strong> 2017-11-21</p>
<p><strong>Code version:</strong> 2362b001a03415fc8138c8fd8298587a49866412</p>
<p><strong>Setting important directories. Also loading important libraries and custom functions for analysis.</strong></p>
<pre class="r"><code>seq_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/data&quot;
file_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/output&quot;
Rdata_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/data&quot;
Script_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/code&quot;
source(file.path(Script_dir,&#39;init.R&#39;))</code></pre>
<p><strong>Untarring big files</strong></p>
<pre class="r"><code>untar(file.path(seq_dir,&quot;genes.fpkm_table.tar.gz&quot;), exdir = seq_dir)
untar(file.path(seq_dir, &quot;isoforms.fpkm_table.tar.gz&quot;), exdir = seq_dir)
untar(file.path(seq_dir, &quot;isoforms.count_table.tar.gz&quot;), exdir = seq_dir)</code></pre>
<p><strong>Loading FPKM datasets generated through cuffquant at the gene and isoform levels</strong></p>
<pre class="r"><code>#Gene FPKMs
fpkms&lt;-read.delim(file.path(seq_dir,&quot;genes.fpkm_table&quot;),row.names=1) # reading in the fpkm table produced by cuffquant
gene_ann&lt;-read.delim(file.path(seq_dir,&quot;genes.attr_table&quot;)) # reading in the gene annotation table
rownames(gene_ann)&lt;-gene_ann$tracking_id #renaming row names

#Isoform FPKMs
isoform_fpkms&lt;-read.delim(file.path(seq_dir,&quot;isoforms.fpkm_table&quot;)) # reading in isoform fpkm table produced from cuffquant
row.names(isoform_fpkms) &lt;- isoform_fpkms$tracking_id # renaming rows
isoform_fpkms &lt;- isoform_fpkms[,-1] # removing the first column</code></pre>
<p><strong>Annotating the single-cell samples and genes</strong></p>
<pre class="r"><code>#Sample Annotation
sample_ann&lt;-read.delim(file.path(seq_dir,&quot;samples.table&quot;)) # reading in sample table
sample_ann$cell_id&lt;-str_extract(as.character(sample_ann$sample_id),&quot;TH_0[0-9]+_[A-H][0-9]+&quot;) #assigning cell IDs to the sample annotation table
rownames(sample_ann)&lt;-sample_ann$cell_id # renaming rows the cell ID names
colnames(fpkms)&lt;-sample_ann$cell_id # renaming columns in fpkms table to the cell ID names

master_cell_sheet&lt;-read.delim(file.path(seq_dir,&quot;CORRECTED_20160605_TH_master_sample_info_sheet.txt&quot;)) # reading in the master cell sheet that contains all pData about cells
master_cell_sheet&lt;-master_cell_sheet[,1:28] # keeping columns 1:28 only

sample_info&lt;-merge(sample_ann,master_cell_sheet,by.x=0,by.y=&#39;cell_id&#39;,sort=FALSE) # merging sample annotation with master_cell_sheet to get all information about cells in 1 place
rownames(sample_info)&lt;-sample_info$cell_id # renaming rows
sample_info&lt;-sample_info[,-1] # getting rid of &quot;Row.names&quot; column created by merging

gene_info&lt;-read.delim(file.path(seq_dir,&quot;gencode.vM8.transcripts.filtered.gene_annotation-eGFP.tab&quot;)) # loading in gencode vM8 gene annotatin for mouse genes
keyCols&lt;-c(&quot;gene_status&quot;,&#39;level&#39;,&#39;gene_type&#39;) # keeping only this information

tmp&lt;-merge(gene_ann,gene_info[,keyCols],by=0,sort=FALSE) # merging gene ann and addition gencode gene information
rownames(tmp)&lt;-tmp$Row.names # renaming rows
tmp&lt;-tmp[,-1] # getting rid of &quot;Row.names&quot; column

gene_ann&lt;-tmp # create final annotation table

lncRNA_geneIds&lt;-gene_info$gene_id[gene_info$gene_type == &quot;lincRNA&quot;] # pulling out all long non-coding RNA genes</code></pre>
<p><strong>Creating the CellDataSet (cds) and removing negative controls/empty wells</strong></p>
<pre class="r"><code>fd&lt;-new(&quot;AnnotatedDataFrame&quot;,data=gene_ann) # officially making fData for genes
pd&lt;-new(&quot;AnnotatedDataFrame&quot;,data=sample_info) # officially making pData for cells

# Creating our new cellDataSet for use in Monocle applications
dat.relative &lt;- newCellDataSet(as(as.matrix(fpkms),&quot;sparseMatrix&quot;),
                      phenoData=pd,
                      featureData=fd)

# Checking the number of cells in the experiment
nrow(pData(dat.relative)) # 480 cells</code></pre>
<pre><code>## [1] 480</code></pre>
<pre class="r"><code># Removing all the cells that are &quot;neg_ctls&quot; and &quot;empty&quot;
dat.relative &lt;- dat.relative[,pData(dat.relative)$type==&quot;sample&quot;]
nrow(pData(dat.relative)) # 473 cells</code></pre>
<pre><code>## [1] 473</code></pre>
<p><strong>Using the Monocle Census algorithm to infer RNA copies per cell and creating a new cds</strong></p>
<pre class="r"><code># Infer RNA copies per cell using relative2abs on isoform-level estimated t with the Census algorithm
fpkm_matrix_adj &lt;- relative2abs(dat.relative, t_estimate=estimate_t(isoform_fpkms),cores=4)

# Create new dat object with absolute RNA copies per cell (estimated).  Allows us to use faster and better characterized negbinom distribution for testing.
fd&lt;-new(&quot;AnnotatedDataFrame&quot;,data=fData(dat.relative))
pd&lt;-new(&quot;AnnotatedDataFrame&quot;,data=pData(dat.relative))

dat &lt;- newCellDataSet(as(as.matrix(fpkm_matrix_adj),&quot;sparseMatrix&quot;),
                      phenoData = pd,
                      featureData = fd,
                      expressionFamily=negbinomial.size(),
                      lowerDetectionLimit=0.1)</code></pre>
<p><strong>Filtering on all cells</strong></p>
<p><strong>Filtering based on number of genes detected in each cell and removing cells that fall outside of determined ranges</strong></p>
<pre class="r"><code># Loading the data
dat.filter &lt;- dat

# &#39;detect&#39; genes and summarize by cell and by gene
dat.filter&lt;-detectGenes(dat.filter,min_expr=0.1)
nrow(pData(dat.filter)) # number of cells 473</code></pre>
<pre><code>## [1] 473</code></pre>
<pre class="r"><code># Plot num genes expressed across cells
hist(pData(dat.filter)$num_genes_expressed,breaks=50,col=&quot;steelblue&quot;,main=&quot;Genes expressed per cell&quot;)</code></pre>
<p><img src="cds_creation-qc_filtering_files/figure-html/filter_by_num_genes_expressed-1.png" width="672" /></p>
<pre class="r"><code># Setting the expressed genes threshold based on the plot above
low.num.genes.threshold &lt;- 2000 # determined by looking at the graph
high.num.genes.threshold &lt;- 10000 # determined by looking at the graph
hist(pData(dat.filter)$num_genes_expressed,
     breaks=50,
     col=&quot;steelblue&quot;,
     main=&quot;Genes expressed per cell&quot;,
     xlim = c(0,10000))
abline(v=low.num.genes.threshold,lty = &quot;dashed&quot;)
abline(v=high.num.genes.threshold,lty=&quot;dashed&quot;)</code></pre>
<p><img src="cds_creation-qc_filtering_files/figure-html/filter_by_num_genes_expressed-2.png" width="672" /></p>
<pre class="r"><code>nrow(pData(dat.filter)) # 473</code></pre>
<pre><code>## [1] 473</code></pre>
<pre class="r"><code># Getting rid of the cells that are outside the thresholds
dat.filter &lt;- dat.filter[,row.names(subset(pData(dat.filter),num_genes_expressed &gt;= low.num.genes.threshold))]
dat.filter &lt;- dat.filter[,row.names(subset(pData(dat.filter),num_genes_expressed &lt;= high.num.genes.threshold))]

# Replotting to check distribution
hist(pData(dat.filter)$num_genes_expressed,
     breaks=50,
     col=&quot;steelblue&quot;,
     main=&quot;Genes expressed per cell&quot;,
     xlim = c(0,10000))
abline(v=low.num.genes.threshold,lty = &quot;dashed&quot;)
abline(v=high.num.genes.threshold,lty=&quot;dashed&quot;)</code></pre>
<p><img src="cds_creation-qc_filtering_files/figure-html/filter_by_num_genes_expressed-3.png" width="672" /></p>
<pre class="r"><code># Number of cells left
nrow(pData(dat.filter)) #431</code></pre>
<pre><code>## [1] 431</code></pre>
<p><strong>Filtering based on the total mass of each cell and removing cells that fall outside of determined ranges</strong></p>
<pre class="r"><code># Plotting the distribution of total cell mass
hist(pData(dat.filter)$total_mass,
     breaks=50,
     col=&quot;darkred&quot;,
     main=&quot;Distribution of Total mass by cell&quot;)

# Setting the threshold based on the graph above
massThreshold&lt;-1e+05 # Adjust this based on observation of mass distribution
massThreshold.2 &lt;-13e+05
hist(pData(dat.filter)$total_mass,
     breaks=50,
     col=&quot;darkred&quot;,
     main=&quot;Distribution of Total mass by cell&quot;)
abline(v=massThreshold,lty=&quot;dashed&quot;)
abline(v=massThreshold.2,lty=&quot;dashed&quot;)</code></pre>
<p><img src="cds_creation-qc_filtering_files/figure-html/Cell%20mass%20threshold%20filtering-1.png" width="672" /></p>
<pre class="r"><code># Filter out cells with &#39;low&#39; number of fragments and &#39;high&#39;
nrow(pData(dat.filter))</code></pre>
<pre><code>## [1] 431</code></pre>
<pre class="r"><code>dat.filter&lt;-dat.filter[,row.names(subset(pData(dat.filter),total_mass &gt;= massThreshold))]
dat.filter&lt;-dat.filter[,row.names(subset(pData(dat.filter),total_mass &lt;= massThreshold.2))]
nrow(pData(dat.filter)) # 410</code></pre>
<pre><code>## [1] 410</code></pre>
<pre class="r"><code># Replot to look at distribution
hist(pData(dat.filter)$total_mass,
     breaks=50,
     col=&quot;darkred&quot;,
     main=&quot;Distribution of Total mass by cell&quot;)
abline(v=massThreshold,lty=&quot;dashed&quot;)
abline(v=massThreshold.2,lty=&quot;dashed&quot;)</code></pre>
<p><img src="cds_creation-qc_filtering_files/figure-html/Cell%20mass%20threshold%20filtering-2.png" width="672" /></p>
<p><strong>Filtering based on the total mRNAs in each cell and removing cells that fall outside of determined ranges</strong></p>
<pre class="r"><code># Calculate total RNA mass per cell
pData(dat.filter)$Total_mRNAs &lt;- colSums(exprs(dat.filter))

# Plot the distribution of total RNA mass per cell
hist(pData(dat.filter)$Total_mRNAs,breaks=50,
     col=&quot;darkgreen&quot;,
     main=&quot;Distribution of mRNA copies per cell&quot;,
     xlim = c(0,5e+04))

# Set high and low thresholds determined from the distribution above
high.numGeneThreshold &lt;- 4e+04
low.num.genes.threshold &lt;- 1000

# Replot distribution with thresholds plotted
hist(pData(dat.filter)$Total_mRNAs,breaks=50,
     col=&quot;darkgreen&quot;,
     main=&quot;Distribution of mRNA copies per cell&quot;,
     xlim = c(0,5e+04))
abline(v=high.numGeneThreshold,lty=&quot;dashed&quot;)
abline(v=low.num.genes.threshold,lty=&quot;dashed&quot;)</code></pre>
<p><img src="cds_creation-qc_filtering_files/figure-html/copies_per_cell_qc-1.png" width="672" /></p>
<pre class="r"><code># Remove cells that fall under the high threshold
nrow(pData(dat.filter)) # 410</code></pre>
<pre><code>## [1] 410</code></pre>
<pre class="r"><code>dat.filter&lt;-dat.filter[,pData(dat.filter)$Total_mRNAs &lt;= high.numGeneThreshold]
nrow(pData(dat.filter)) # 410</code></pre>
<pre><code>## [1] 410</code></pre>
<p><strong>Now that only high quality cells are retained, the detectGenes function is used to recalculate the # of genes detected in each cell</strong></p>
<pre class="r"><code># Recalculate # of genes
dat.filter&lt;-detectGenes(dat.filter,min_expr=0.1)</code></pre>
<p><strong>Before beginning Monocle analysis, the data needs to be checked to ensure that it follows a log-normal distribution. We find that our data follows a log-normal distribution. Script is pulled directly from Monocle manual.</strong></p>
<pre class="r"><code># Log-transform each value in the expression matrix.
L &lt;- log(exprs(dat.filter))
# Standardize each gene, so that they are all on the same scale, Then melt
# the data with plyr so we can plot it easily
melted_dens_df &lt;- melt(t(scale(t(L))))
# Plot the distribution of the standardized gene expression values.
qplot(value, geom = &quot;density&quot;, data = melted_dens_df) + stat_function(fun = dnorm,
size = 0.5, color = &quot;red&quot;) + xlab(&quot;Standardized log(Transcripts)&quot;) + ylab(&quot;Density&quot;)</code></pre>
<pre><code>## Warning: Removed 11850640 rows containing non-finite values (stat_density).</code></pre>
<p><img src="cds_creation-qc_filtering_files/figure-html/Checking%20to%20see%20if%20final%20expression%20matrix%20is%20log-normal-1.png" width="672" /></p>
<pre class="r"><code>## Warning: Removed 11850640 rows containing non-finite values (stat_density).</code></pre>
<p><strong>The data is subsetted according to age and region. These subsetted cds are saved as .rds objects to be used in subsequent analyses</strong></p>
<pre class="r"><code>nrow(pData(dat.filter)) # 410 cells</code></pre>
<pre><code>## [1] 410</code></pre>
<pre class="r"><code>saveRDS(object = dat.filter, file = file.path(Rdata_dir,&quot;dat.filter.rds&quot;))</code></pre>
<pre class="r"><code>e15.dat.filter&lt;-dat.filter[,pData(dat.filter)$age==&quot;E15.5&quot;]
nrow(pData(e15.dat.filter)) # 172 cells at e15 before filtering</code></pre>
<pre><code>## [1] 172</code></pre>
<pre class="r"><code>saveRDS(object = e15.dat.filter, file = file.path(Rdata_dir,&quot;e15.dat.filter.rds&quot;))</code></pre>
<pre class="r"><code>e15.Mb.dat.filter &lt;- dat.filter[,pData(dat.filter)$age==&quot;E15.5&quot; &amp; pData(dat.filter)$region==&quot;MB&quot;]
nrow(pData(e15.Mb.dat.filter)) # 88 cells</code></pre>
<pre><code>## [1] 88</code></pre>
<pre class="r"><code>saveRDS(object = e15.Mb.dat.filter, file = file.path(Rdata_dir,&quot;e15.Mb.dat.filter.rds&quot;))</code></pre>
<pre class="r"><code>e15.Fb.dat.filter &lt;- dat.filter[,pData(dat.filter)$age==&quot;E15.5&quot; &amp; pData(dat.filter)$region==&quot;FB&quot;]
nrow(pData(e15.Fb.dat.filter)) # 84 cells</code></pre>
<pre><code>## [1] 84</code></pre>
<pre class="r"><code>saveRDS(object = e15.Fb.dat.filter, file = file.path(Rdata_dir,&quot;e15.Fb.dat.filter.rds&quot;))</code></pre>
<pre class="r"><code>P7.dat.filter&lt;-dat.filter[,pData(dat.filter)$age==&quot;P7&quot;]
nrow(pData(P7.dat.filter)) # 238 cells at P7 before filtering</code></pre>
<pre><code>## [1] 238</code></pre>
<pre class="r"><code>saveRDS(object = P7.dat.filter, file = file.path(Rdata_dir,&quot;P7.dat.filter.rds&quot;))</code></pre>
<pre class="r"><code>P7.Mb.dat.filter&lt;-dat.filter[,pData(dat.filter)$age==&quot;P7&quot; &amp; pData(dat.filter)$region==&quot;MB&quot;]
nrow(pData(P7.Mb.dat.filter)) # 80</code></pre>
<pre><code>## [1] 80</code></pre>
<pre class="r"><code>saveRDS(object = P7.Mb.dat.filter, file = file.path(Rdata_dir, &quot;P7.Mb.dat.filter.rds&quot;))</code></pre>
<pre class="r"><code>P7.Fb.dat.filter&lt;-dat.filter[,pData(dat.filter)$age==&quot;P7&quot; &amp; pData(dat.filter)$region==&quot;FB&quot;]
nrow(pData(P7.Fb.dat.filter)) # 84</code></pre>
<pre><code>## [1] 84</code></pre>
<pre class="r"><code>saveRDS(object = P7.Fb.dat.filter, file = file.path(Rdata_dir, &quot;P7.Fb.dat.filter.rds&quot;))</code></pre>
<pre class="r"><code>P7.Ob.dat.filter&lt;-dat.filter[,pData(dat.filter)$age==&quot;P7&quot; &amp; pData(dat.filter)$region==&quot;OB&quot;]
nrow(pData(P7.Ob.dat.filter)) # 74</code></pre>
<pre><code>## [1] 74</code></pre>
<pre class="r"><code>saveRDS(object = P7.Ob.dat.filter, file = file.path(Rdata_dir, &quot;P7.Ob.dat.filter.rds&quot;))</code></pre>
<p><strong>Removing uncompressed large files before commit</strong></p>
<pre class="r"><code>file.remove(file.path(seq_dir, &quot;isoforms.fpkm_table&quot;))</code></pre>
<pre><code>## [1] TRUE</code></pre>
<pre class="r"><code>file.remove(file.path(seq_dir, &quot;genes.fpkm_table&quot;))</code></pre>
<pre><code>## [1] TRUE</code></pre>
<pre class="r"><code>file.remove(file.path(seq_dir, &quot;isoforms.count_table&quot;))</code></pre>
<pre><code>## [1] TRUE</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.3.0 (2016-05-03)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X 10.11.6 (El Capitan)
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
##  [1] grid      splines   stats4    parallel  stats     graphics  grDevices
##  [8] utils     datasets  methods   base     
## 
## other attached packages:
##  [1] ggbiplot_0.55       scales_0.5.0        SC3_1.1.4          
##  [4] ROCR_1.0-7          jackstraw_1.1.1     lfa_1.2.2          
##  [7] tsne_0.1-3          gridExtra_2.3       slackr_1.4.2       
## [10] vegan_2.4-4         permute_0.9-4       MASS_7.3-47        
## [13] gplots_3.0.1        RColorBrewer_1.1-2  Hmisc_4.0-3        
## [16] Formula_1.2-2       survival_2.41-3     lattice_0.20-35    
## [19] Heatplus_2.18.0     Rtsne_0.13          pheatmap_1.0.8     
## [22] tidyr_0.7.1         dplyr_0.7.4         plyr_1.8.4         
## [25] heatmap.plus_1.3    stringr_1.2.0       marray_1.50.0      
## [28] limma_3.28.21       reshape2_1.4.2      monocle_2.2.0      
## [31] DDRTree_0.1.5       irlba_2.2.1         VGAM_1.0-2         
## [34] ggplot2_2.2.1       Biobase_2.32.0      BiocGenerics_0.18.0
## [37] Matrix_1.2-11      
## 
## loaded via a namespace (and not attached):
##  [1] RSelenium_1.7.1        colorspace_1.3-2       class_7.3-14          
##  [4] rprojroot_1.2          htmlTable_1.9          corpcor_1.6.9         
##  [7] base64enc_0.1-3        mvtnorm_1.0-6          codetools_0.2-15      
## [10] doParallel_1.0.11      robustbase_0.92-7      knitr_1.17            
## [13] jsonlite_1.5           cluster_2.0.6          semver_0.2.0          
## [16] shiny_1.0.5            rrcov_1.4-3            httr_1.3.1            
## [19] backports_1.1.1        assertthat_0.2.0       lazyeval_0.2.1        
## [22] acepack_1.4.1          htmltools_0.3.6        tools_3.3.0           
## [25] bindrcpp_0.2           igraph_1.1.2           gtable_0.2.0          
## [28] glue_1.1.1             binman_0.1.0           doRNG_1.6.6           
## [31] Rcpp_0.12.13           slam_0.1-37            gdata_2.18.0          
## [34] nlme_3.1-131           iterators_1.0.8        mime_0.5              
## [37] rngtools_1.2.4         gtools_3.5.0           WriteXLS_4.0.0        
## [40] XML_3.98-1.9           DEoptimR_1.0-8         yaml_2.1.14           
## [43] pkgmaker_0.22          rpart_4.1-11           fastICA_1.2-1         
## [46] latticeExtra_0.6-28    stringi_1.1.5          pcaPP_1.9-72          
## [49] foreach_1.4.3          e1071_1.6-8            checkmate_1.8.4       
## [52] caTools_1.17.1         rlang_0.1.2            pkgconfig_2.0.1       
## [55] matrixStats_0.52.2     bitops_1.0-6           qlcMatrix_0.9.5       
## [58] evaluate_0.10.1        purrr_0.2.3            bindr_0.1             
## [61] labeling_0.3           htmlwidgets_0.9        magrittr_1.5          
## [64] R6_2.2.2               combinat_0.0-8         wdman_0.2.2           
## [67] foreign_0.8-69         mgcv_1.8-22            nnet_7.3-12           
## [70] tibble_1.3.4           KernSmooth_2.23-15     rmarkdown_1.6         
## [73] data.table_1.10.4      HSMMSingleCell_0.106.2 digest_0.6.12         
## [76] xtable_1.8-2           httpuv_1.3.5           openssl_0.9.7         
## [79] munsell_0.4.3          registry_0.3</code></pre>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
