<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Paul Hook" />


<title>E15.5 Analysis</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/simplex.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-0.9/htmlwidgets.js"></script>
<script src="site_libs/datatables-binding-0.2/datatables.js"></script>
<link href="site_libs/dt-core-1.10.12/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.12/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.12/js/jquery.dataTables.min.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 41px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 46px;
  margin-top: -46px;
}

.section h2 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h3 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h4 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h5 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h6 {
  padding-top: 46px;
  margin-top: -46px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">sc-da-parkinsons</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-line-chart"></span>
     
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="cds_creation-qc_filtering.html">CDS creation and QC</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="E15.clustering.html">E15.5 Analyses</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="P7.analyses.links.html">P7 Analyses</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="All.cells.analyses.links.html">All cell analyses</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="PD.GWAS.scoring.links.html">PD Gene Scoring</a>
    </li>
  </ul>
</li>
<li>
  <a href="paper-figures.html">
    <span class="fa fa-paperclip"></span>
     
    Paper Figures
  </a>
</li>
<li>
  <a href="https://github.com/pwh124/sc-da-parkinsons">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">E15.5 Analysis</h1>
<h4 class="author"><em>Paul Hook</em></h4>

</div>


<p><strong>Last update:</strong> 2017-12-15</p>
<p><strong>Code version:</strong> 97cdbba92f549b083d57cc1acd1fcd954ca4c492</p>
<div id="setting-important-directories.-also-loading-important-libraries-and-custom-functions-for-analysis." class="section level3">
<h3>Setting important directories. Also loading important libraries and custom functions for analysis.</h3>
<pre class="r"><code>seq_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/data&quot;
file_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/output&quot;
Rdata_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/data&quot;
Script_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/code&quot;
source(file.path(Script_dir,&#39;init.R&#39;))
source(file.path(Script_dir,&quot;tools_R.r&quot;))</code></pre>
</div>
<div id="loading-the-e15.5-all-cell-.rds" class="section level3">
<h3>Loading the E15.5 all cell .rds</h3>
<pre class="r"><code>e15.dat.filter &lt;- readRDS(file = file.path(Rdata_dir, &quot;e15.dat.filter.final.Rds&quot;))
nrow(pData(e15.dat.filter)) # 172</code></pre>
<pre><code>## [1] 172</code></pre>
</div>
<div id="performing-differentially-expression" class="section level3">
<h3>Performing differentially expression</h3>
<pre class="r"><code>e15.Diff.Test &lt;- differentialGeneTest(e15.dat.filter, 
                             fullModelFormulaStr = &quot;~subset.cluster&quot;,
                             reducedModelFormulaStr = &quot;~1&quot;,
                             relative_expr = FALSE,
                             cores=8,
                             verbose = F)


nrow(e15.Diff.Test)</code></pre>
<pre><code>## [1] 14409</code></pre>
<pre class="r"><code># Only keeping differentially expressed genes with a FDF &lt; 5%
e15.tmp&lt;-e15.Diff.Test[e15.Diff.Test$qval &lt; 0.05,]
# Ordering by q-value
e15.tmp.2 &lt;- e15.tmp[order(e15.tmp$qval),]
e15.Diff.Genes &lt;- as.character(e15.tmp.2$gene_short_name)
length(e15.Diff.Genes) # 1196</code></pre>
<pre><code>## [1] 1196</code></pre>
</div>
<div id="t-sne-plots-of-top-10-differentially-expressed-genes" class="section level3">
<h3>t-SNE plots of top 10 differentially expressed genes</h3>
<pre class="r"><code>q &lt;-myTSNEPlotAlpha(e15.dat.filter,color=&quot;subset.cluster&quot;, shape=&quot;age&quot;,markers = e15.Diff.Genes[1:10],scaled = T) + scale_color_brewer(palette=&quot;Set1&quot;) + ggtitle(&quot;e15 Mb - BCV tSNE - Top 10 Differentially Expressed Genes&quot;)</code></pre>
<pre><code>## Scale for &#39;colour&#39; is already present. Adding another scale for
## &#39;colour&#39;, which will replace the existing scale.</code></pre>
<pre class="r"><code>q</code></pre>
<p><img src="e15.cluster.specific.genes_files/figure-html/Making%20t-SNE-1.png" width="672" /></p>
</div>
<div id="finding-specific-genes-for-each-cluster" class="section level3">
<h3>Finding specific genes for each cluster</h3>
<pre class="r"><code>e15.dat.diff.filter &lt;- e15.dat.filter[row.names(e15.tmp.2),]
nrow(e15.dat.diff.filter)</code></pre>
<pre><code>## Features 
##     1196</code></pre>
<pre class="r"><code>subset.clusters &lt;- levels(pData(e15.dat.diff.filter)$subset.cluster)
mean.clusters &lt;- data.frame(rownames(exprs(e15.dat.diff.filter)))
percent.clusters &lt;- data.frame(rownames(exprs(e15.dat.diff.filter)))

for(i in 1:length(subset.clusters)){
  tmp.dat &lt;- e15.dat.diff.filter[,pData(e15.dat.diff.filter)$subset.cluster == subset.clusters[i]]

  per.tmp&lt;-as.data.frame(apply(exprs(tmp.dat),1, function(x) length(which(x &gt; 1))/ncol(tmp.dat)))
  colnames(per.tmp) &lt;- paste0(subset.clusters[i],&quot;.perc&quot;) 

  mean.tmp&lt;- as.data.frame(apply(log2(exprs(tmp.dat)+1),1,mean))
  mean.tmp &lt;- as.data.frame(mean.tmp)
  colnames(mean.tmp) &lt;- paste0(subset.clusters[i],&quot;.mean&quot;)
  
  mean.clusters &lt;- merge(x = mean.clusters, y = mean.tmp, by.x = &quot;rownames.exprs.e15.dat.diff.filter..&quot;, by.y = 0)
  percent.clusters &lt;- merge(x = percent.clusters, y = per.tmp, by.x = &quot;rownames.exprs.e15.dat.diff.filter..&quot;, by.y = 0)
}

rownames(mean.clusters) &lt;- mean.clusters$rownames.exprs.e15.dat.diff.filter..
mean.clusters &lt;- mean.clusters[,-1]
mean.clusters &lt;- as.matrix(mean.clusters)

dat.JSD &lt;- .specificity(mat = mean.clusters,logMode = F, relative = F)
dat.JSD &lt;- as.data.frame(dat.JSD)

rownames(percent.clusters) &lt;- percent.clusters$rownames.exprs.e15.dat.diff.filter..
percent.clusters &lt;- percent.clusters[,-1]

merged.df &lt;- merge(x = dat.JSD, y = percent.clusters, by = 0)
row.names(merged.df) &lt;- merged.df$Row.names
merged.df &lt;- merged.df[,-1]</code></pre>
<pre class="r"><code>color &lt;- c(&quot;#F6937A&quot;,&quot;#882F1C&quot;,&quot;#BF1F36&quot;,&quot;#C2B280&quot;)
color.transparent &lt;- adjustcolor(color, alpha.f = 0.3)
bins &lt;- seq(from = 0, to = 1, by= 0.005)

hist(dat.JSD$E15.FB.1.mean, breaks = bins, col = color.transparent[1], main = &quot;Specificity Distributions&quot;, ylab = &quot;Genes&quot;, xlab = &quot;Specificity score&quot;)
hist(dat.JSD$E15.FB.2.mean, breaks = bins, add = T, col = color.transparent[2])
hist(dat.JSD$E15.MB.1.mean, breaks = bins, add = T, col = color.transparent[3])
hist(dat.JSD$E15.MB.2.mean, breaks = bins, add = T, col = color.transparent[4])
abline(v = 0.4, lwd = 2)</code></pre>
<p><img src="e15.cluster.specific.genes_files/figure-html/Determining%20the%20specificity%20score%20to%20use%20as%20a%20cutoff-1.png" width="672" /></p>
<pre class="r"><code>e15.specific.genes &lt;- data.frame()

for(i in 1:4){
  gene.tmp.1 &lt;- merged.df[merged.df[,i] &gt;= 0.4 &amp; merged.df[,i+4] &gt;= 0.4,]
  gene.ordered &lt;- row.names(gene.tmp.1[order(-gene.tmp.1[,i]),])
  cluster.genes &lt;- data.frame(lookupGeneName(e15.dat.filter, gene.ordered))
  e15.specific.genes &lt;- cbindPad(e15.specific.genes, cluster.genes)
}

names(e15.specific.genes) &lt;- c(&quot;E15.FB.1&quot;,&quot;E15.FB.2&quot;,&quot;E15.MB.1&quot;,&quot;E15.MB.2&quot;)

write.table(e15.specific.genes, file = file.path(file_dir,&quot;E15.specific.genes.txt&quot;), sep = &quot;\t&quot;, quote = F, row.names = F)</code></pre>
</div>
<div id="making-an-interactive-table-for-specific-genes" class="section level3">
<h3>Making an interactive table for specific genes</h3>
<pre class="r"><code>library(DT)
datatable(e15.specific.genes)</code></pre>
<pre><code>## Warning in instance$preRenderHook(instance): It seems your data is too
## big for client-side DataTables. You may consider server-side processing:
## http://rstudio.github.io/DT/server.html</code></pre>
<div id="htmlwidget-e0c2e2742a5382bafd08" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e0c2e2742a5382bafd08">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60"],["Dlx2","Dlx1","Dlx5","Fam84a","Gria1","Gabra5","Igfbpl1","Rnd3","Fgf14","Pnoc","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""],["Gm5532","Npy","Six3os1","Gas1","Six3","Eepd1","Map3k1","Esrrg","Rpp25","Prox1","Fign","Sp8","Gm29374","Isl1","Sst","A830082K12Rik","Ecel1","Zeb2","Nr2f1","Slc44a5","Nr2f2","Gpc3","2900055J20Rik","Kcnmb2","Tmem132b","Zbtb7c","Unc5c","Nefm","Lrrtm1","Pde2a","Tenm3","Zfp804a","Galnt13","Hs3st5","Anks1","1110015O18Rik","Ubash3b","Tle1","Sema3a","Tshz2","Klhl5","Kcnma1","Zdhhc2","Gad2","Meis2","Klf12","Gad1","AI593442","Lrrc4c","Gm14204","Islr2","Tmem2","Id4","Cygb","Lhfp","Grik2","Rnf220","Rftn1","Vcan",""],["Pax5","Lhx9","Ebf1","Npnt","Tspan9","Nrn1","Nrg1","Gabra1","Sorcs2","Slc17a6","Mab21l1","Pcdh18","Syn2","Car10","Ppp1r14c","Fam19a2","Tmem163","Meis1","Rasgrp2","Cacna2d1","Pde4b","Lrrtm3","Wscd1","Surf6","Adap1","Mir99ahg","Pcdh17","Tmem91","Garnl3","Ckmt1","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""],["Foxa1","Lmx1a","C130021I20Rik","Slc6a3","Chrna6","Pitx3","Foxa2","Gch1","Ntn1","En1","Schip1","Nr4a2","eGFP","Ret","Ddc","Slc18a2","Necab1","Wnt5a","Ctnna1","Th","Cpne7","Cnpy1","Syt17","Ptpru","Kcnip4","Uncx","Dlk1","Ebf2","Slc10a4","Foxp2","Adarb2","Klhl1","Serpine2","Pdzrn4","Igfbp5","Bnc2","Wdr54","St6galnac4","Tox2","Snca","Gm2694","Calb2","Tcf12","Rasgef1b","2810025M15Rik","Slc1a2","Cdk14","Sdc2","Pbx1","Syt13","Mt3","Ifitm2","Rit2","Mpped2","Slit2","Epha5","Epdr1","Ptprg","Osbpl5","Stk19"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>E15.FB.1<\/th>\n      <th>E15.FB.2<\/th>\n      <th>E15.MB.1<\/th>\n      <th>E15.MB.2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]},"selection":{"mode":"multiple","selected":null,"target":"row"}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.3.0 (2016-05-03)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X 10.11.6 (El Capitan)
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
##  [1] grid      splines   stats4    parallel  stats     graphics  grDevices
##  [8] utils     datasets  methods   base     
## 
## other attached packages:
##  [1] DT_0.2              ggbiplot_0.55       scales_0.5.0       
##  [4] SC3_1.1.4           ROCR_1.0-7          jackstraw_1.1.1    
##  [7] lfa_1.2.2           tsne_0.1-3          gridExtra_2.3      
## [10] slackr_1.4.2        vegan_2.4-4         permute_0.9-4      
## [13] MASS_7.3-47         gplots_3.0.1        RColorBrewer_1.1-2 
## [16] Hmisc_4.0-3         Formula_1.2-2       survival_2.41-3    
## [19] lattice_0.20-35     Heatplus_2.18.0     Rtsne_0.13         
## [22] pheatmap_1.0.8      tidyr_0.7.1         dplyr_0.7.4        
## [25] plyr_1.8.4          heatmap.plus_1.3    stringr_1.2.0      
## [28] marray_1.50.0       limma_3.28.21       reshape2_1.4.2     
## [31] monocle_2.2.0       DDRTree_0.1.5       irlba_2.2.1        
## [34] VGAM_1.0-2          ggplot2_2.2.1       Biobase_2.32.0     
## [37] BiocGenerics_0.18.0 Matrix_1.2-11      
## 
## loaded via a namespace (and not attached):
##  [1] RSelenium_1.7.1        colorspace_1.3-2       class_7.3-14          
##  [4] rprojroot_1.2          htmlTable_1.9          corpcor_1.6.9         
##  [7] base64enc_0.1-3        mvtnorm_1.0-6          codetools_0.2-15      
## [10] doParallel_1.0.11      robustbase_0.92-7      knitr_1.17            
## [13] jsonlite_1.5           cluster_2.0.6          semver_0.2.0          
## [16] shiny_1.0.5            rrcov_1.4-3            httr_1.3.1            
## [19] backports_1.1.1        assertthat_0.2.0       lazyeval_0.2.1        
## [22] acepack_1.4.1          htmltools_0.3.6        tools_3.3.0           
## [25] bindrcpp_0.2           igraph_1.1.2           gtable_0.2.0          
## [28] glue_1.1.1             binman_0.1.0           doRNG_1.6.6           
## [31] Rcpp_0.12.13           slam_0.1-37            gdata_2.18.0          
## [34] nlme_3.1-131           iterators_1.0.8        mime_0.5              
## [37] rngtools_1.2.4         gtools_3.5.0           WriteXLS_4.0.0        
## [40] XML_3.98-1.9           DEoptimR_1.0-8         yaml_2.1.15           
## [43] pkgmaker_0.22          rpart_4.1-11           fastICA_1.2-1         
## [46] latticeExtra_0.6-28    stringi_1.1.5          pcaPP_1.9-72          
## [49] foreach_1.4.3          e1071_1.6-8            checkmate_1.8.4       
## [52] caTools_1.17.1         rlang_0.1.2            pkgconfig_2.0.1       
## [55] matrixStats_0.52.2     bitops_1.0-6           qlcMatrix_0.9.5       
## [58] evaluate_0.10.1        purrr_0.2.3            bindr_0.1             
## [61] labeling_0.3           htmlwidgets_0.9        magrittr_1.5          
## [64] R6_2.2.2               combinat_0.0-8         wdman_0.2.2           
## [67] foreign_0.8-69         mgcv_1.8-22            nnet_7.3-12           
## [70] tibble_1.3.4           KernSmooth_2.23-15     rmarkdown_1.8         
## [73] data.table_1.10.4      HSMMSingleCell_0.106.2 digest_0.6.12         
## [76] xtable_1.8-2           httpuv_1.3.5           openssl_0.9.7         
## [79] munsell_0.4.3          registry_0.3</code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-318359-10"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-318359-10');
</script>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
