<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Paul Hook" />


<title>Figure.S6.Rmd</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/simplex.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 41px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 46px;
  margin-top: -46px;
}

.section h2 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h3 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h4 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h5 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h6 {
  padding-top: 46px;
  margin-top: -46px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">sc-da-parkinsons</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-line-chart"></span>
     
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="cds_creation-qc_filtering.html">CDS creation and QC</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="E15.clustering.html">E15.5 Analyses</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="P7.analyses.links.html">P7 Analyses</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="All.cells.analyses.links.html">All cell analyses</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="PD.GWAS.scoring.links.html">PD Gene Scoring</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="Cplx1_Th-striatum_analysis.html">Cplx1 Th+ striatum density</a>
    </li>
  </ul>
</li>
<li>
  <a href="paper-figures.html">
    <span class="fa fa-paperclip"></span>
     
    Paper Figures
  </a>
</li>
<li>
  <a href="shiny-link.html">
    <span class="fa fa-bar-chart"></span>
     
    Explore Data
  </a>
</li>
<li>
  <a href="https://github.com/pwh124/sc-da-parkinsons">
    <span class="fa fa-github"></span>
     
    GitHub Repo
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Figure.S6.Rmd</h1>
<h4 class="author"><em>Paul Hook</em></h4>

</div>


<hr />
<p><strong>Last update:</strong> 2018-02-02</p>
<p><strong>Code version:</strong> 03314b1d9d662f91341e7902bc398b5c80c294f1</p>
<hr />
<div id="setting-important-directories" class="section level3">
<h3>Setting important directories</h3>
<p>Also loading important libraries and custom functions for analysis.</p>
<pre class="r"><code>seq_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/data&quot;
file_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/output&quot;
Rdata_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/data&quot;
Script_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/code&quot;
source(file.path(Script_dir,&#39;init.R&#39;))
source(file.path(Script_dir,&quot;tools_R.r&quot;))

library(pryr)

## Add an alpha value to a colour
add.alpha &lt;- function(col, alpha=1){
  if(missing(col))
    stop(&quot;Please provide a vector of colours.&quot;)
  apply(sapply(col, col2rgb)/255, 2, 
                     function(x) 
                       rgb(x[1], x[2], x[3], alpha=alpha))  
}</code></pre>
</div>
<div id="loading-and-summarizing-the-data" class="section level2">
<h2>Loading and summarizing the data</h2>
<p>First we need to load the data into R. The gene-SNP data was compiled from Ensembl, using NHGRI GWAS catalog SNPs with +/- 1 Mb coordinates.</p>
<pre class="r"><code># Loading gene-SNP-data complied from Ensembl
dat &lt;- read.delim(file.path(file_dir,&quot;gwas.snp_results.txt&quot;))

# Changing SNP column to factor
dat$snp &lt;- as.factor(dat$snp)

# Check on how many SNPs we have
length(levels(dat$snp)) </code></pre>
<pre><code>## [1] 40213</code></pre>
<div id="adding-mouse-homolog-data" class="section level3">
<h3>Adding mouse homolog data</h3>
<p>In order to do any simulation, we need to add mouse homologs to the mix. I used a modified version of the Homologene database from MGI, which only contained genes with 1-to-1 mouse-human homologs. I then will merge the tables and remove genes with no 1-to-1 mouse homolog, much like a did when calculated numbers for our analyses.</p>
<pre class="r"><code># First need to load and process the mouse MGI data
# Processed in the &quot;biomart.finding.genes.Rmd&quot;
homolog.table &lt;- read.delim(file.path(file_dir,&quot;one-one.homologs.txt&quot;))

# Merge these tables in order to add mouse genes to dat os they can be scored
dat.merge &lt;- unique(merge(dat, homolog.table, by.x = &quot;hgnc_symbol&quot;, by.y = &quot;HumanSymbol&quot;, all.x = T))

# Check for duplicated rows
summary(duplicated(dat.merge))</code></pre>
<pre><code>##    Mode   FALSE    NA&#39;s 
## logical 1308796       0</code></pre>
<pre class="r"><code># Check number of unique SNPs queried
length(levels(dat.merge$snp))</code></pre>
<pre><code>## [1] 40213</code></pre>
<pre class="r"><code># Remove those without mouse homologs
filtered.dat.merge &lt;- dat.merge[!is.na(dat.merge$MouseSymbol),]

# Check number of unique SNPs again
length(levels(filtered.dat.merge$snp))</code></pre>
<pre><code>## [1] 40213</code></pre>
</div>
<div id="score-genes-for-expression" class="section level3">
<h3>Score genes for expression</h3>
<p>Now that I have a final dataset to work with, I can score the genes based on expression in our SN DA cluster as well as from the La Manno SN DA cluster.</p>
<pre class="r"><code># Loading in the SN expression data SNC expressed genes in our data
SN.expressed.genes &lt;- readRDS(file = file.path(Rdata_dir, &quot;SN.expressed.genes.rds&quot;))

# SNC expresed genes in Linnarsson Data
lin.snc.exprs &lt;- readRDS(file = file.path(Rdata_dir, &quot;lin.snc.expressed.rds&quot;))

# Scoring based on expression
filtered.dat.merge$expression &lt;- as.factor(if_else(filtered.dat.merge$MouseSymbol %in% 
    SN.expressed.genes, 1, 0))

# Scoring based on being expressed in Lin
filtered.dat.merge$lin.exprs &lt;- as.factor(if_else(filtered.dat.merge$MouseSymbol %in% 
    lin.snc.exprs, 1, 0))</code></pre>
</div>
<div id="summarizing-data" class="section level3">
<h3>Summarizing data</h3>
<p>Now that every gene in every locus is scored for expression, the data needed to be summarized by locus. Data will be summarized at the level of total genes with 1:1 human:mouse homologs in the locus as well as by how many genes are expressed in each locus</p>
<pre class="r"><code># Summarizing
dat.merge.sum &lt;- filtered.dat.merge %&gt;%
  dplyr::group_by(snp) %&gt;%
  dplyr::summarise(total.genes = n(),
                   exprs.genes = sum(expression == 1 | lin.exprs == 1))</code></pre>
<pre><code>## Warning: package &#39;bindrcpp&#39; was built under R version 3.3.2</code></pre>
</div>
<div id="figure-s6a" class="section level3">
<h3>Figure S6A</h3>
<p>A permutation needed to be set up that would randomly sample 49 loci from the summarized dataset we have (dat.merge.sum) and then compute the frequency of genes in those randomly sampled loci that are expressed. This simulation would give us some idea as to whether the percentage of expressed genes in SN DA neurons in PD GWAS loci is similar to that of random GWAS loci. Since the data appeared to be normal, the 43% figure in our data was tested for signficance.</p>
<pre class="r"><code># Setting up blank vectors to be populated
count.freq &lt;- c()
total.freq &lt;- c()
exprs.freq &lt;- c()

# Running my for loop
set.seed(24)
for(i in 1:10000){
  temp &lt;- sample_n(dat.merge.sum,49) # sampling 49 loci
  total &lt;- sum(temp$total.genes) # Getting total number of genes in that sampling
  total.freq &lt;- c(total,total.freq) # Collecting that number
  exprs &lt;- sum(temp$exprs.genes) # Getting total number of expressed genes in that sampling
  exprs.freq &lt;- c(exprs, exprs.freq) # Collecting that number
  freq &lt;- exprs/total # Calculating percentage of expressed genes
  count.freq &lt;- c(freq, count.freq) # Collecting that number
  i = i + 1 # Moving the counter forward
}

# Looking at normality of the data
qqnorm(count.freq)
qqline(count.freq, col=2)</code></pre>
<p><img src="Figure.S6_files/figure-html/Figure%206A-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Looking at the significance of 0.43
p.test &lt;- pnorm(0.43,mean=mean(count.freq),sd=sd(count.freq), lower.tail = F)

p1 %&lt;a-% {
  hist(count.freq, breaks=50,
     xlab=&quot;Percentage of genes expressed in 49 loci&quot;,
     main=&quot;&quot;,
     ylab = &quot;Number of simulations&quot;,
     col = &quot;#2E8B5766&quot;)
  abline(v=0.43,lwd=2,col=&quot;red&quot;)
  text(c(0.455,0.455),
      c(900,930),
      labels = c(paste0(&quot;P = &quot;,round(p.test,5)),
                  paste0(&quot;0.43&quot;)), col = &quot;red&quot;, font = 2)
}

pdf(file.path(file_dir,&quot;Figure.6A.pdf&quot;),height = 3, width = 2.5, pointsize = 8, family = &#39;Helvetica&#39;)
p1
dev.off()</code></pre>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
<pre class="r"><code>p1</code></pre>
<p><img src="Figure.S6_files/figure-html/Figure%206A-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-s6b" class="section level3">
<h3>Figure S6B</h3>
<p>A permutation was run to see whether or not we are seeing a greater than expected number of loci with &gt;= 1 SN expressed gene in our PD GWAS loci when compared to random GWAS loci. It does look like the majority of permutations have less than 49 loci (to the left of red line) with at least 1 gene expressed.</p>
<pre class="r"><code># Setting up blank vector to be populated
count.1 &lt;- c()
# Running my for loop
set.seed(24)
for(i in 1:10000){
  temp &lt;- sample_n(dat.merge.sum,49) # randomly sample 49 loci
  tf &lt;- temp$exprs.genes &gt;= 1 # Create a logical vector 
  num &lt;- length(tf[tf==TRUE]) # Count loci wtih &gt;= 1 gene expressed
  count.1 &lt;- c(num, count.1) # Collect that number
  i = i + 1 # Move the counter
}

# Plotting
p2 %&lt;a-% {
  hist(count.1, breaks=10,
     xlab = &quot;Number of loci out of 49&quot;,
     ylab = &quot;Number of simulations&quot;,
     main = &quot;&quot;,
     col=&quot;#6A5ACD66&quot;)
  abline(v=48, lwd=2, col=&quot;red&quot;)
  text(&quot;49&quot;,col=&#39;red&#39;,y=750,x=49)
  arrows(x0=48,x1=49,y0=2600,y1=2600, col=&#39;red&#39;,lwd=2)
}

pdf(file.path(file_dir,&quot;one.gene_loci_simulation.pdf&quot;), height = 3, width = 2.5, pointsize = 8, family = &#39;Helvetica&#39;)
p2
dev.off()</code></pre>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
<pre class="r"><code>p2</code></pre>
<p><img src="Figure.S6_files/figure-html/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-s6c" class="section level3">
<h3>Figure S6C</h3>
<p>A permutation was run in order to see if the number of PD GWAS loci with <em>only</em> one SN DA neuron expressed gene that we find in our paper (n = 3) is different than what we would expect from 49 random GWAS loci. We see that only finding 3/49 loci wtih 1 gene expressed is actually lower than what would be expected from random loci.</p>
<pre class="r"><code># Set up blank vector
count &lt;- c()
# Running for loop
set.seed(24)
for(i in 1:10000){
  temp &lt;- sample_n(dat.merge.sum,49) # sampling 49 loci
  tf &lt;- temp$exprs.genes == 1 # Making on a logical vector
  num &lt;- length(tf[tf==TRUE]) # counting number of loci wtih exactly 1 gene expressed
  count &lt;- c(num, count) # Collecting that number
  i = i + 1 # Move that counter!
}

p3 %&lt;a-% {
  hist(count, breaks=20,
       xlab = &quot;Number of loci out of 49&quot;,
       ylab = &quot;Number of simulations&quot;,
       main = &quot;&quot;,
       col = add.alpha(&quot;sienna3&quot;,alpha=0.4))
  abline(v=3, lwd=2, col=&quot;red&quot;)
  text(&quot;n = 3&quot;,x=1,y=1500, font = 2, col = &#39;red&#39;)
}

#main = &quot;Simulation of NHGRI GWAS loci\nwith *only* 1 gene expressed in SN DA neurons&quot;

pdf(file.path(file_dir,&quot;only.one_simulation.pdf&quot;), width = 3, height = 2.5, pointsize = 8, family = &#39;Helvetica&#39;)
p3
dev.off()</code></pre>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
<pre class="r"><code>p3</code></pre>
<p><img src="Figure.S6_files/figure-html/Figure%20S6C-1.png" width="672" /></p>
</div>
<div id="session-info" class="section level3">
<h3>Session Info</h3>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.3.0 (2016-05-03)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X 10.11.6 (El Capitan)
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
##  [1] grid      splines   stats4    parallel  stats     graphics  grDevices
##  [8] utils     datasets  methods   base     
## 
## other attached packages:
##  [1] bindrcpp_0.2        pryr_0.1.3          ggbiplot_0.55      
##  [4] scales_0.5.0        SC3_1.1.4           ROCR_1.0-7         
##  [7] jackstraw_1.1.1     lfa_1.2.2           tsne_0.1-3         
## [10] gridExtra_2.3       slackr_1.4.2        vegan_2.4-4        
## [13] permute_0.9-4       MASS_7.3-47         gplots_3.0.1       
## [16] RColorBrewer_1.1-2  Hmisc_4.0-3         Formula_1.2-2      
## [19] survival_2.41-3     lattice_0.20-35     Heatplus_2.18.0    
## [22] Rtsne_0.13          pheatmap_1.0.8      tidyr_0.7.1        
## [25] dplyr_0.7.4         plyr_1.8.4          heatmap.plus_1.3   
## [28] stringr_1.2.0       marray_1.50.0       limma_3.28.21      
## [31] reshape2_1.4.3      monocle_2.2.0       DDRTree_0.1.5      
## [34] irlba_2.2.1         VGAM_1.0-2          ggplot2_2.2.1      
## [37] Biobase_2.32.0      BiocGenerics_0.18.0 Matrix_1.2-11      
## 
## loaded via a namespace (and not attached):
##  [1] RSelenium_1.7.1        colorspace_1.3-2       class_7.3-14          
##  [4] rprojroot_1.2          htmlTable_1.9          corpcor_1.6.9         
##  [7] base64enc_0.1-3        mvtnorm_1.0-6          codetools_0.2-15      
## [10] doParallel_1.0.11      robustbase_0.92-7      knitr_1.17            
## [13] jsonlite_1.5           cluster_2.0.6          semver_0.2.0          
## [16] shiny_1.0.5            rrcov_1.4-3            httr_1.3.1            
## [19] backports_1.1.1        assertthat_0.2.0       lazyeval_0.2.1        
## [22] formatR_1.5            acepack_1.4.1          htmltools_0.3.6       
## [25] tools_3.3.0            igraph_1.1.2           gtable_0.2.0          
## [28] glue_1.1.1             binman_0.1.0           doRNG_1.6.6           
## [31] Rcpp_0.12.14           slam_0.1-37            gdata_2.18.0          
## [34] nlme_3.1-131           iterators_1.0.8        mime_0.5              
## [37] rngtools_1.2.4         gtools_3.5.0           WriteXLS_4.0.0        
## [40] XML_3.98-1.9           DEoptimR_1.0-8         yaml_2.1.15           
## [43] pkgmaker_0.22          rpart_4.1-11           fastICA_1.2-1         
## [46] latticeExtra_0.6-28    stringi_1.1.5          pcaPP_1.9-72          
## [49] foreach_1.4.3          e1071_1.6-8            checkmate_1.8.4       
## [52] caTools_1.17.1         rlang_0.1.6            pkgconfig_2.0.1       
## [55] matrixStats_0.52.2     bitops_1.0-6           qlcMatrix_0.9.5       
## [58] evaluate_0.10.1        purrr_0.2.4            bindr_0.1             
## [61] htmlwidgets_0.9        magrittr_1.5           R6_2.2.2              
## [64] combinat_0.0-8         wdman_0.2.2            foreign_0.8-69        
## [67] mgcv_1.8-22            nnet_7.3-12            tibble_1.3.4          
## [70] KernSmooth_2.23-15     rmarkdown_1.8          data.table_1.10.4     
## [73] HSMMSingleCell_0.106.2 digest_0.6.12          xtable_1.8-2          
## [76] httpuv_1.3.5           openssl_0.9.7          munsell_0.4.3         
## [79] registry_0.3</code></pre>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-318359-10"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-318359-10');
</script>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
