<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Paul Hook" />


<title>P7 MB Recursive Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/simplex.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 41px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 46px;
  margin-top: -46px;
}

.section h2 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h3 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h4 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h5 {
  padding-top: 46px;
  margin-top: -46px;
}
.section h6 {
  padding-top: 46px;
  margin-top: -46px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">sc-da-parkinsons</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-line-chart"></span>
     
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="cds_creation-qc_filtering.html">CDS creation and QC</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="E15.clustering.html">E15.5 Analyses</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="P7.analyses.links.html">P7 Analyses</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="All.cells.analyses.links.html">All cell analyses</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="PD.GWAS.scoring.links.html">PD Gene Scoring</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="Cplx1_Th-striatum_analysis.html">Cplx1 Th+ striatum density</a>
    </li>
  </ul>
</li>
<li>
  <a href="paper-figures.html">
    <span class="fa fa-paperclip"></span>
     
    Paper Figures
  </a>
</li>
<li>
  <a href="shiny-link.html">
    <span class="fa fa-bar-chart"></span>
     
    Explore Data
  </a>
</li>
<li>
  <a href="https://github.com/pwh124/sc-da-parkinsons">
    <span class="fa fa-github"></span>
     
    GitHub Repo
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">P7 MB Recursive Analysis</h1>
<h4 class="author"><em>Paul Hook</em></h4>

</div>


<p><strong>Last update:</strong> 2018-01-18</p>
<p><strong>Code version:</strong> 5996c19850dc5fc65723f30bc33ab1c8d083a7e6</p>
<div id="setting-important-directories" class="section level3">
<h3>Setting important directories</h3>
<p>Also loading important libraries and custom functions for analysis.</p>
<pre class="r"><code>seq_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/data&quot;
file_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/output&quot;
Rdata_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/data&quot;
Script_dir &lt;- &quot;/Volumes/PAULHOOK/sc-da-parkinsons/code&quot;
source(file.path(Script_dir,&#39;init.R&#39;))
source(file.path(Script_dir,&quot;tools_R.r&quot;))</code></pre>
</div>
<div id="loading-mb-.rds" class="section level3">
<h3>Loading MB .rds</h3>
<pre class="r"><code>P7.Mb.dat.filter &lt;- readRDS(file = file.path(Rdata_dir, &quot;P7.Mb.dat.filter.rds&quot;))</code></pre>
</div>
<div id="filter-genes-by-percentage-of-cells-expresssing-each-gene" class="section level3">
<h3>Filter genes by percentage of cells expresssing each gene</h3>
<pre class="r"><code># Plot number of cells expressing each gene as histogram
hist(fData(P7.Mb.dat.filter)$num_cells_expressed,breaks=100,col=&quot;red&quot;,main=&quot;Cells expressed per gene&quot;)</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/filter%20by%20cells%20expressing%20each%20gene-1.png" width="672" /></p>
<pre class="r"><code># Keep only expressed genes with expression in &gt;= 5% of cells
numCellThreshold&lt;-nrow(pData(P7.Mb.dat.filter))*0.05
P7.Mb.dat.expressed_genes&lt;-row.names(subset(fData(P7.Mb.dat.filter),num_cells_expressed &gt;= numCellThreshold))

# Same plot as above with threshold
hist(fData(P7.Mb.dat.filter)$num_cells_expressed,breaks=100,col=&quot;red&quot;,main=&quot;Cells expressed per gene - threshold&quot;)
abline(v=numCellThreshold,lty=&quot;dashed&quot;)</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/filter%20by%20cells%20expressing%20each%20gene-2.png" width="672" /></p>
</div>
<div id="prepping-the-monocle-model-for-analysis" class="section level3">
<h3>Prepping the Monocle model for analysis</h3>
<pre class="r"><code># Only keeping &quot;expressed&quot; genes
P7.Mb.dat.filter &lt;-P7.Mb.dat.filter[P7.Mb.dat.expressed_genes,]

# Estimating the size factors
P7.Mb.dat.filter &lt;-estimateSizeFactors(P7.Mb.dat.filter)

# Estimating dispersions
P7.Mb.dat.filter &lt;- estimateDispersions(P7.Mb.dat.filter,cores=8)</code></pre>
<pre><code>## Warning: Deprecated, use tibble::rownames_to_column() instead.</code></pre>
<pre><code>## Removing 144 outliers</code></pre>
<pre class="r"><code># Removing 144 outliers
# Warning message:
# Deprecated, use tibble::rownames_to_column() instead. </code></pre>
</div>
<div id="calculating-summary-stats" class="section level3">
<h3>Calculating summary stats</h3>
<pre class="r"><code># Calculating summary stats
fData(P7.Mb.dat.filter)$mean_expr&lt;-apply(round(exprs(P7.Mb.dat.filter)),1,mean) # mean expression
fData(P7.Mb.dat.filter)$sd_expr&lt;-apply(round(exprs(P7.Mb.dat.filter)),1,sd) # sd expression
fData(P7.Mb.dat.filter)$bcv&lt;-(fData(P7.Mb.dat.filter)$sd_expr/fData(P7.Mb.dat.filter)$mean_expr)**2 # calculating biological coefficient of variation
fData(P7.Mb.dat.filter)$percent_detection&lt;-(fData(P7.Mb.dat.filter)$num_cells_expressed/dim(P7.Mb.dat.filter)[2])*100 # calculating % detection</code></pre>
</div>
<div id="identifying-high-dispersion-genes" class="section level3">
<h3>Identifying high dispersion genes</h3>
<pre class="r"><code>P7.Mb.dat.filter.genes &lt;- P7.Mb.dat.filter # spoofing the CellDataSet
disp_table &lt;- dispersionTable(P7.Mb.dat.filter.genes) # pulling out the dispersion table
unsup_clustering_genes &lt;-subset(disp_table, mean_expression &gt;= 0.5 &amp; dispersion_empirical &gt;= 1.5 * dispersion_fit) # subsetting the data to pull out genes with expression above 0.5 and dispersion empirical &gt; 2
P7.Mb.dat.high_bcv_genes&lt;-unsup_clustering_genes$gene_id # pulling out list of genes
P7.Mb.dat.filter.order &lt;- setOrderingFilter(P7.Mb.dat.filter, unsup_clustering_genes$gene_id)
plot_ordering_genes(P7.Mb.dat.filter.order) # plotting the dispersion and genes</code></pre>
<pre><code>## Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/high_dispersion_genes_monocle%20-%20Pulling%20out%20the%20high%20dispersion%20genes%20for%20PCA%20analysis-1.png" width="672" /></p>
<pre class="r"><code>length(P7.Mb.dat.high_bcv_genes) # 923</code></pre>
<pre><code>## [1] 923</code></pre>
</div>
<div id="running-pca-with-high-dispersion-genes" class="section level3">
<h3>Running PCA with high dispersion genes</h3>
<pre class="r"><code># BCV Identified high dispersion genes. Running PC analysis
P7.Mb.dat.filter.BCV.pca&lt;-prcomp(t(log2(exprs(P7.Mb.dat.filter[P7.Mb.dat.high_bcv_genes,])+1)),center=T,scale. = TRUE)

# Plotting the PCA graphs
# Plotting the first 2 PCs and coloring by age
hvPCA1&lt;-ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$age,ellipse=T,var.axes=F) + scale_color_manual(values=c(&quot;darkgreen&quot;,&quot;red&quot;)) + monocle:::monocle_theme_opts()

# Plotting the first 2 PCs and coloring by region
hvPCA2&lt;-ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$region,ellipse=T,var.axes=F) + scale_color_brewer(palette=&quot;Set1&quot;) + monocle:::monocle_theme_opts() + ggtitle(&quot;P7 MB PCA&quot;)

# Plotting the first 2 PCs and coloring by plate the cell was sequenced from
hvPCA3&lt;-ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$split_plate,ellipse=T, var.axes=F) + scale_color_brewer(palette=&quot;Set1&quot;) + monocle:::monocle_theme_opts()

# Show the plots in the terminal
hvPCA1</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/Run%20PCA%20with%20high%20BCV%20genes-1.png" width="672" /></p>
<pre class="r"><code>hvPCA2</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/Run%20PCA%20with%20high%20BCV%20genes-2.png" width="672" /></p>
<pre class="r"><code>hvPCA3</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/Run%20PCA%20with%20high%20BCV%20genes-3.png" width="672" /></p>
</div>
<div id="outlier-identification" class="section level3">
<h3>Outlier identification</h3>
<p>The first pass analysis seems to show obvious outliers in the PCA. Below, the relevant data is saved to look at outliers in the future</p>
<pre class="r"><code>MB.outlier.plot &lt;- ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$region,ellipse=F,var.axes=F)

saveRDS(MB.outlier.plot, file = file.path(Rdata_dir,&quot;P7.MB.PCA.outliers.rds&quot;))

saveRDS(hvPCA2, file = file.path(Rdata_dir,&quot;P7.MB.PCA.outliers.rds&quot;))</code></pre>
</div>
<div id="outlier-analysis" class="section level3">
<h3>Outlier analysis</h3>
<p>In an attempt to identify what the outliers are, genes driving the outlier appearance are extracted and saved for GSEA analysis</p>
<pre class="r"><code># PC1 seems to describe outliers in the data
P7.MB.dat.pca.rotations.df &lt;- as.data.frame(P7.Mb.dat.filter.BCV.pca$rotation[,1:6])
genes.tmp &lt;- P7.MB.dat.pca.rotations.df[order(-P7.MB.dat.pca.rotations.df$PC1),]
PC1.genes &lt;- as.character(row.names(genes.tmp))
PC1.genes.names &lt;- lookupGeneName(P7.Mb.dat.filter,PC1.genes)
genes.tmp$gene_short_name &lt;- PC1.genes.names

PC1.genes.df &lt;- genes.tmp[,c(7,1)]
write.table(PC1.genes.df, file = file.path(file_dir, &quot;P7.Mb.cells.PC1.rnk&quot;),quote = F, sep = &#39;\t&#39;, row.names = F, col.names = F)

# PC2
P7.MB.dat.pca.rotations.df &lt;- as.data.frame(P7.Mb.dat.filter.BCV.pca$rotation[,1:6])
genes.tmp &lt;- P7.MB.dat.pca.rotations.df[order(-P7.MB.dat.pca.rotations.df$PC2),]
PC2.genes &lt;- as.character(row.names(genes.tmp))
PC2.genes.names &lt;- lookupGeneName(P7.Mb.dat.filter,PC2.genes)
genes.tmp$gene_short_name &lt;- PC2.genes.names

PC2.genes.df &lt;- genes.tmp[,c(7,2)]
write.table(PC2.genes.df, file = file.path(file_dir, &quot;P7.Mb.cells.PC2.rnk&quot;),quote = F, sep = &#39;\t&#39;, row.names = F, col.names = F)</code></pre>
</div>
<div id="removing-outliers-from-the-original-mb-dataset" class="section level3">
<h3>Removing outliers from the original MB dataset</h3>
<pre class="r"><code>pData(P7.Mb.dat.filter)$PC1 &lt;- P7.Mb.dat.filter.BCV.pca$x[,1]
pData(P7.Mb.dat.filter)$PC2 &lt;- P7.Mb.dat.filter.BCV.pca$x[,2]
P7.Mb.dat.outliers.1&lt;-as.character(pData(P7.Mb.dat.filter[,pData(P7.Mb.dat.filter)$PC2 &gt;= 15])$sample_id.x)
P7.Mb.dat.outliers.2&lt;-as.character(pData(P7.Mb.dat.filter[,pData(P7.Mb.dat.filter)$PC1 &gt;= 15])$sample_id.x)
P7.Mb.outliers &lt;- c(P7.Mb.dat.outliers.1, P7.Mb.dat.outliers.2)
P7.Mb.outliers &lt;- unique(P7.Mb.outliers)
length(P7.Mb.outliers)</code></pre>
<pre><code>## [1] 5</code></pre>
<pre class="r"><code>P7.Mb.dat.filter &lt;- readRDS(file = file.path(Rdata_dir, &quot;P7.Mb.dat.filter.rds&quot;))

P7.Mb.dat.filter &lt;- P7.Mb.dat.filter[,!(pData(P7.Mb.dat.filter)$sample_id.x %in% P7.Mb.outliers)]

nrow(pData(P7.Mb.dat.filter)) #75</code></pre>
<pre><code>## [1] 75</code></pre>
<pre class="r"><code>saveRDS(object = P7.Mb.outliers, file = file.path(Rdata_dir,&quot;P7.Mb.outliers.rds&quot;))</code></pre>
</div>
<div id="rerunning-analysis-on-p7-mb-with-outliers-removed." class="section level3">
<h3>Rerunning analysis on P7 MB with outliers removed.</h3>
<p>Starting here, all work performed above was re-run with the outliers removed.</p>
</div>
<div id="filter-genes-by-percentage-of-cells-expresssing-each-gene-with-outliers-removed" class="section level3">
<h3>Filter genes by percentage of cells expresssing each gene with outliers removed</h3>
<pre class="r"><code># Plot number of cells expressing each gene as histogram
hist(fData(P7.Mb.dat.filter)$num_cells_expressed,breaks=100,col=&quot;red&quot;,main=&quot;Cells expressed per gene&quot;)</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/filter%20by%20cells%20expressing%20each%20gene%20with%20outliers%20removed-1.png" width="672" /></p>
<pre class="r"><code># Keep only expressed genes with expression in &gt;= 5% of cells
numCellThreshold&lt;-nrow(pData(P7.Mb.dat.filter))*0.05
P7.Mb.dat.expressed_genes&lt;-row.names(subset(fData(P7.Mb.dat.filter),num_cells_expressed &gt;= numCellThreshold))

# Same plot as above with threshold
hist(fData(P7.Mb.dat.filter)$num_cells_expressed,breaks=100,col=&quot;red&quot;,main=&quot;Cells expressed per gene - threshold&quot;)
abline(v=numCellThreshold,lty=&quot;dashed&quot;)</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/filter%20by%20cells%20expressing%20each%20gene%20with%20outliers%20removed-2.png" width="672" /></p>
</div>
<div id="prepping-the-monocle-model-for-analysis-with-outliers-removed" class="section level3">
<h3>Prepping the Monocle model for analysis with outliers removed</h3>
<pre class="r"><code># Only keeping &quot;expressed&quot; genes
P7.Mb.dat.filter &lt;-P7.Mb.dat.filter[P7.Mb.dat.expressed_genes,]

# Estimating the size factors
P7.Mb.dat.filter &lt;-estimateSizeFactors(P7.Mb.dat.filter)

# Estimating dispersions
P7.Mb.dat.filter &lt;- estimateDispersions(P7.Mb.dat.filter,cores=8)</code></pre>
<pre><code>## Warning: Deprecated, use tibble::rownames_to_column() instead.</code></pre>
<pre><code>## Removing 221 outliers</code></pre>
<pre class="r"><code># Removing 221 outliers
# Warning message:
# Deprecated, use tibble::rownames_to_column() instead. </code></pre>
</div>
<div id="calculating-summary-stats-with-outliers-removed" class="section level3">
<h3>Calculating summary stats with outliers removed</h3>
<pre class="r"><code># Calculating summary stats
fData(P7.Mb.dat.filter)$mean_expr&lt;-apply(round(exprs(P7.Mb.dat.filter)),1,mean) # mean expression
fData(P7.Mb.dat.filter)$sd_expr&lt;-apply(round(exprs(P7.Mb.dat.filter)),1,sd) # sd expression
fData(P7.Mb.dat.filter)$bcv&lt;-(fData(P7.Mb.dat.filter)$sd_expr/fData(P7.Mb.dat.filter)$mean_expr)**2 # calculating biological coefficient of variation
fData(P7.Mb.dat.filter)$percent_detection&lt;-(fData(P7.Mb.dat.filter)$num_cells_expressed/dim(P7.Mb.dat.filter)[2])*100 # calculating % detection</code></pre>
</div>
<div id="identifying-high-dispersion-genes-with-outliers-removed" class="section level3">
<h3>Identifying high dispersion genes with outliers removed</h3>
<pre class="r"><code>P7.Mb.dat.filter.genes &lt;- P7.Mb.dat.filter # spoofing the CellDataSet
disp_table &lt;- dispersionTable(P7.Mb.dat.filter.genes) # pulling out the dispersion table
unsup_clustering_genes &lt;-subset(disp_table, mean_expression &gt;= 0.5 &amp; dispersion_empirical &gt;= 1.5 * dispersion_fit) # subsetting the data to pull out genes with expression above 0.5 and dispersion empirical &gt; 2
P7.Mb.dat.high_bcv_genes&lt;-unsup_clustering_genes$gene_id # pulling out list of genes
P7.Mb.dat.filter.order &lt;- setOrderingFilter(P7.Mb.dat.filter, unsup_clustering_genes$gene_id)
plot_ordering_genes(P7.Mb.dat.filter.order) # plotting the dispersion and genes</code></pre>
<pre><code>## Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/high_dispersion_genes_monocle%20-%20Pulling%20out%20the%20high%20dispersion%20genes%20for%20PCA%20analysis%20with%20outliers%20removed-1.png" width="672" /></p>
<pre class="r"><code>length(P7.Mb.dat.high_bcv_genes) # 1185</code></pre>
<pre><code>## [1] 1185</code></pre>
</div>
<div id="running-pca-with-high-dispersion-genes-with-outliers-removed" class="section level3">
<h3>Running PCA with high dispersion genes with outliers removed</h3>
<pre class="r"><code># BCV Identified high dispersion genes. Running PC analysis
P7.Mb.dat.filter.BCV.pca&lt;-prcomp(t(log2(exprs(P7.Mb.dat.filter[P7.Mb.dat.high_bcv_genes,])+1)),center=T,scale. = TRUE)

# Plotting the PCA graphs
# Plotting the first 2 PCs and coloring by age
hvPCA1&lt;-ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$age,ellipse=T,var.axes=F) + scale_color_manual(values=c(&quot;darkgreen&quot;,&quot;red&quot;)) + monocle:::monocle_theme_opts()

# Plotting the first 2 PCs and coloring by region
hvPCA2&lt;-ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$region,ellipse=T,var.axes=F) + scale_color_brewer(palette=&quot;Set1&quot;) + monocle:::monocle_theme_opts()

# Plotting the first 2 PCs and coloring by plate the cell was sequenced from
hvPCA3&lt;-ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$split_plate,ellipse=T,var.axes=F) + scale_color_brewer(palette=&quot;Set1&quot;) + monocle:::monocle_theme_opts()

# Show the plots in the terminal
hvPCA1</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/Run%20PCA%20with%20high%20BCV%20genes%20with%20outliers%20removed-1.png" width="672" /></p>
<pre class="r"><code>hvPCA2</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/Run%20PCA%20with%20high%20BCV%20genes%20with%20outliers%20removed-2.png" width="672" /></p>
<pre class="r"><code>hvPCA3</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/Run%20PCA%20with%20high%20BCV%20genes%20with%20outliers%20removed-3.png" width="672" /></p>
<pre class="r"><code># No real outliers seen</code></pre>
</div>
<div id="screeplots" class="section level3">
<h3>Screeplots</h3>
<p>Viewing screeplots and determining the number of “significant” PCs. Since no additional outliers were identified in the PCA plot above, we will continue with the analysis with only the original outliers removed</p>
<pre class="r"><code># Making a screeplot of the BCV PCA. This will help determine how many
# principal components we should use in our tSNE visualization Show this
# plot
screeplot(P7.Mb.dat.filter.BCV.pca, npcs = 30, main = &quot;P7 MB - High BCV Genes PCA Screeplot&quot;)
abline(v = 5.5, lwd = 2, col = &quot;red&quot;)</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/Screeplots%20of%20Data-1.png" width="672" /></p>
<pre class="r"><code>ggscreeplot(P7.Mb.dat.filter.BCV.pca, type = &quot;pev&quot;)</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/Screeplots%20of%20Data-2.png" width="672" /></p>
<pre class="r"><code># Conclustion: Seems to be clear that just the first 5 PCs explain the most
# variation in our data</code></pre>
</div>
<div id="creating-a-t-sne-plot-from-the-significant-pcs-with-outliers-removed" class="section level3">
<h3>Creating a t-SNE plot from the “significant” PCs with outliers removed</h3>
<pre class="r"><code>nComponents&lt;-5 # estimated from the screeplots
#seed &lt;- runif(1,1,9999) # determined by testing random seeds
seed &lt;- 7641.869
set.seed(seed) #setting seed

P7.Mb.dat.filter.BCV.tsne&lt;-tsne(P7.Mb.dat.filter.BCV.pca$x[,1:nComponents],perplexity=10,max_iter=5000,whiten = FALSE)

pData(P7.Mb.dat.filter)$tSNE1_pos&lt;-P7.Mb.dat.filter.BCV.tsne[,1]
pData(P7.Mb.dat.filter)$tSNE2_pos&lt;-P7.Mb.dat.filter.BCV.tsne[,2]

P7.Mb.dat.filter.BCV.tsne.plot&lt;-myTSNEPlotAlpha(P7.Mb.dat.filter,color=&quot;region&quot;, shape=&quot;age&quot;) + scale_color_brewer(palette=&quot;Set1&quot;) + ggtitle(&quot;P7 Mb - BCV tSNE Plot&quot;)

P7.Mb.dat.filter.BCV.tsne.plot</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/tsne%20with%20outliers%20removed-1.png" width="672" /></p>
</div>
<div id="identifying-clusters" class="section level3">
<h3>Identifying clusters</h3>
<p>Identifying clusters in the data in an unsupervised manner with outliers removed</p>
<pre class="r"><code># Going to attempt to use the R program &#39;ADPclust&#39; to determine how many
# clusters our data has

# Loading NbClust
library(ADPclust)

# Running ADPclust
clust.res &lt;- adpclust(x = P7.Mb.dat.filter.BCV.tsne)

# Extracting the &#39;best partition&#39; (aka the best cluster) for each cell
clust.res.df &lt;- as.data.frame(clust.res$cluster)

# Adding the cluster assignment for each cell to the pData
pData(P7.Mb.dat.filter)$kmeans_tSNE_cluster &lt;- as.factor(clust.res.df$`clust.res$cluster`)

# Plotting the same tSNE plot as above but coloring with the &#39;clusters&#39;
myTSNEPlotAlpha(P7.Mb.dat.filter, color = &quot;kmeans_tSNE_cluster&quot;, shape = &quot;age&quot;) + 
    scale_color_brewer(palette = &quot;Set1&quot;) + ggtitle(&quot;P7 Mb - BCV tSNE with Clusters Plot&quot;)</code></pre>
<pre><code>## Scale for &#39;colour&#39; is already present. Adding another scale for
## &#39;colour&#39;, which will replace the existing scale.</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/-%20attempting%20to%20cluster-1.png" width="672" /></p>
</div>
<div id="qc-on-the-clusters-with-outliers-removed" class="section level3">
<h3>QC on the clusters with outliers removed</h3>
<pre class="r"><code># Distribution of number of genes expressed in clusters
q &lt;- ggplot(pData(P7.Mb.dat.filter)) +
  geom_density(aes(x=num_genes_expressed,fill=kmeans_tSNE_cluster),alpha=0.3) + scale_fill_brewer(palette=&quot;Set1&quot;) + facet_grid(.~age) + monocle:::monocle_theme_opts()

q</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/check%20the%20clusters%20-%20qc%20on%20clusters%20with%20outliers%20removed-1.png" width="672" /></p>
<pre class="r"><code># Plotting the distribution of total mRNAs in clusters
q&lt;-ggplot(pData(P7.Mb.dat.filter)) +
  geom_density(aes(x=Total_mRNAs,fill=kmeans_tSNE_cluster),alpha=0.3) + scale_fill_brewer(palette=&quot;Set1&quot;) + facet_grid(.~age) + monocle:::monocle_theme_opts()

q</code></pre>
<p><img src="P7_Mb.recursive_analysis_files/figure-html/check%20the%20clusters%20-%20qc%20on%20clusters%20with%20outliers%20removed-2.png" width="672" /></p>
</div>
<div id="saving-the-p7-mb-final-cds" class="section level3">
<h3>Saving the P7 MB final cds</h3>
<pre class="r"><code>P7.Mb.dat.filter.final &lt;- P7.Mb.dat.filter
saveRDS(object = P7.Mb.dat.filter.final, file = file.path(Rdata_dir, &quot;P7.Mb.dat.filter.final.Rds&quot;))</code></pre>
</div>
<div id="extracting-pdata-information-and-saving-it" class="section level3">
<h3>Extracting pData information and saving it</h3>
<pre class="r"><code>P7.Mb.clusters.df &lt;- pData(P7.Mb.dat.filter)[,c(1,39)]
saveRDS(P7.Mb.clusters.df, file = file.path(Rdata_dir, &quot;P7.Mb.clusters.df.rds&quot;))</code></pre>
</div>
<div id="extracting-sn-expressed-genes" class="section level3">
<h3>Extracting SN expressed genes</h3>
<p>In order to score the PD GWAS genes later, we determined the genes expressed in SN DA neurons</p>
<pre class="r"><code># Extracting just the SN DA neurons
sn_exprs &lt;- P7.Mb.dat.filter[,pData(P7.Mb.dat.filter)$kmeans_tSNE_cluster == 4]

# Finding the average expression for each gene
sn.average &lt;- as.data.frame(apply(exprs(sn_exprs),1, function(x) mean(x)))
names(sn.average) &lt;- &quot;average&quot;

# Adding a column so dataframe functions can be used
sn.average$average.2 &lt;- sn.average$average

# Only keeping genes that have an an average &gt;= 0.5 transcripts
sn.expressed &lt;- sn.average[which(sn.average$average &gt;= 0.5),]

# Looking up gene names
sn.expressed$gene &lt;- lookupGeneName(P7.Mb.dat.filter, row.names(sn.expressed))

#Counting number of rowns
nrow(sn.expressed)</code></pre>
<pre><code>## [1] 6127</code></pre>
<pre class="r"><code>#Kepping only unique gene names
SN.expressed.genes &lt;- unique(sn.expressed$gene)

#Saving the rds for later scoring
saveRDS(SN.expressed.genes, file = file.path(Rdata_dir,&quot;SN.expressed.genes.rds&quot;))</code></pre>
</div>
<div id="session-info" class="section level3">
<h3>Session Info</h3>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.3.0 (2016-05-03)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X 10.11.6 (El Capitan)
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
##  [1] grid      splines   stats4    parallel  stats     graphics  grDevices
##  [8] utils     datasets  methods   base     
## 
## other attached packages:
##  [1] ADPclust_0.7        ggbiplot_0.55       scales_0.5.0       
##  [4] SC3_1.1.4           ROCR_1.0-7          jackstraw_1.1.1    
##  [7] lfa_1.2.2           tsne_0.1-3          gridExtra_2.3      
## [10] slackr_1.4.2        vegan_2.4-4         permute_0.9-4      
## [13] MASS_7.3-47         gplots_3.0.1        RColorBrewer_1.1-2 
## [16] Hmisc_4.0-3         Formula_1.2-2       survival_2.41-3    
## [19] lattice_0.20-35     Heatplus_2.18.0     Rtsne_0.13         
## [22] pheatmap_1.0.8      tidyr_0.7.1         dplyr_0.7.4        
## [25] plyr_1.8.4          heatmap.plus_1.3    stringr_1.2.0      
## [28] marray_1.50.0       limma_3.28.21       reshape2_1.4.3     
## [31] monocle_2.2.0       DDRTree_0.1.5       irlba_2.2.1        
## [34] VGAM_1.0-2          ggplot2_2.2.1       Biobase_2.32.0     
## [37] BiocGenerics_0.18.0 Matrix_1.2-11      
## 
## loaded via a namespace (and not attached):
##  [1] RSelenium_1.7.1        colorspace_1.3-2       class_7.3-14          
##  [4] rprojroot_1.2          htmlTable_1.9          corpcor_1.6.9         
##  [7] base64enc_0.1-3        mvtnorm_1.0-6          codetools_0.2-15      
## [10] doParallel_1.0.11      robustbase_0.92-7      knitr_1.17            
## [13] jsonlite_1.5           cluster_2.0.6          semver_0.2.0          
## [16] shiny_1.0.5            rrcov_1.4-3            httr_1.3.1            
## [19] backports_1.1.1        assertthat_0.2.0       lazyeval_0.2.1        
## [22] formatR_1.5            acepack_1.4.1          htmltools_0.3.6       
## [25] tools_3.3.0            bindrcpp_0.2           igraph_1.1.2          
## [28] gtable_0.2.0           glue_1.1.1             binman_0.1.0          
## [31] doRNG_1.6.6            Rcpp_0.12.14           slam_0.1-37           
## [34] gdata_2.18.0           nlme_3.1-131           iterators_1.0.8       
## [37] mime_0.5               rngtools_1.2.4         gtools_3.5.0          
## [40] WriteXLS_4.0.0         XML_3.98-1.9           DEoptimR_1.0-8        
## [43] yaml_2.1.15            pkgmaker_0.22          rpart_4.1-11          
## [46] fastICA_1.2-1          latticeExtra_0.6-28    stringi_1.1.5         
## [49] pcaPP_1.9-72           foreach_1.4.3          e1071_1.6-8           
## [52] checkmate_1.8.4        caTools_1.17.1         rlang_0.1.6           
## [55] pkgconfig_2.0.1        matrixStats_0.52.2     bitops_1.0-6          
## [58] qlcMatrix_0.9.5        evaluate_0.10.1        purrr_0.2.4           
## [61] bindr_0.1              labeling_0.3           htmlwidgets_0.9       
## [64] magrittr_1.5           R6_2.2.2               combinat_0.0-8        
## [67] wdman_0.2.2            foreign_0.8-69         mgcv_1.8-22           
## [70] nnet_7.3-12            tibble_1.3.4           KernSmooth_2.23-15    
## [73] rmarkdown_1.8          data.table_1.10.4      HSMMSingleCell_0.106.2
## [76] digest_0.6.12          xtable_1.8-2           httpuv_1.3.5          
## [79] openssl_0.9.7          munsell_0.4.3          registry_0.3</code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-318359-10"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-318359-10');
</script>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
