---
title: "P7 Analysis"
author: "Paul Hook"
date: "6/21/2017"
output: html_document
---

# Setting important directories. Also loading important libraries and custom functions for analysis.
```{r init,echo=FALSE,message=FALSE, error=FALSE, warning=FALSE, tidy=TRUE}
seq_dir <- "/Volumes/PAULHOOK/GitHub/DA_scRNA-seq/Data"
file_dir <- "/Volumes/PAULHOOK/GitHub/DA_scRNA-seq/file.output"
Rdata_dir <- "/Volumes/PAULHOOK/GitHub/DA_scRNA-seq/rds"
Script_dir <- "/Volumes/PAULHOOK/GitHub/DA_scRNA-seq/Scripts"
source(file.path(Script_dir,'init.R'))
source(file.path(Script_dir,"tools_R.r"))
```

```{r Loading in all P7 cell data}

P7.dat.filter <- readRDS(file = file.path(Rdata_dir, "P7.dat.filter.final.Rds"))
nrow(pData(P7.dat.filter)) # 224

```

```{r Differential expression}

P7.Diff.Test <- differentialGeneTest(P7.dat.filter, 
                             fullModelFormulaStr = "~subset.cluster",
                             reducedModelFormulaStr = "~1",
                             relative_expr = FALSE,
                             cores=8,
                             verbose = F)


nrow(P7.Diff.Test)
# Only keeping differentially expressed genes with a FDF < 5%
P7.tmp<-P7.Diff.Test[P7.Diff.Test$qval < 0.05,]
# Ordering by q-value
P7.tmp.2 <- P7.tmp[order(P7.tmp$qval),]
P7.Diff.Genes <- as.character(P7.tmp.2$gene_short_name)
length(P7.Diff.Genes) # 3971
saveRDS(P7.Diff.Genes, file = "/Volumes/PAULHOOK/GitHub/DA_scRNA-seq/rds/P7.diff.genes.rds")

myTSNEPlotAlpha(P7.dat.filter,color="subset.cluster", shape="age",markers = P7.Diff.Genes[1:20],scaled = T) + scale_color_brewer(palette="Set1") + ggtitle("e15 Mb - BCV tSNE - Top 10 Differentially Expressed Genes")
```

```{r finding specific genes}
P7.dat.diff.filter <- P7.dat.filter[row.names(P7.tmp.2),]
nrow(P7.dat.diff.filter)

subset.clusters <- levels(pData(P7.dat.diff.filter)$subset.cluster)
mean.clusters <- data.frame(rownames(exprs(P7.dat.diff.filter)))
percent.clusters <- data.frame(rownames(exprs(P7.dat.diff.filter)))

for(i in 1:length(subset.clusters)){
  tmp.dat <- P7.dat.diff.filter[,pData(P7.dat.diff.filter)$subset.cluster == subset.clusters[i]]

  per.tmp<-as.data.frame(apply(exprs(tmp.dat),1, function(x) length(which(x > 1))/ncol(tmp.dat)))
  colnames(per.tmp) <- paste0(subset.clusters[i],".perc") 

  mean.tmp<- as.data.frame(apply(log2(exprs(tmp.dat)+1),1,mean))
  mean.tmp <- as.data.frame(mean.tmp)
  colnames(mean.tmp) <- paste0(subset.clusters[i],".mean")
  
  mean.clusters <- merge(x = mean.clusters, y = mean.tmp, by.x = "rownames.exprs.P7.dat.diff.filter..", by.y = 0)
  percent.clusters <- merge(x = percent.clusters, y = per.tmp, by.x = "rownames.exprs.P7.dat.diff.filter..", by.y = 0)
}

rownames(mean.clusters) <- mean.clusters$rownames.exprs.P7.dat.diff.filter..
mean.clusters <- mean.clusters[,-1]
mean.clusters <- as.matrix(mean.clusters)

dat.JSD <- .specificity(mat = mean.clusters,logMode = F, relative = F)
dat.JSD <- as.data.frame(dat.JSD)

rownames(percent.clusters) <- percent.clusters$rownames.exprs.P7.dat.diff.filter..
percent.clusters <- percent.clusters[,-1]

merged.df <- merge(x = dat.JSD, y = percent.clusters, by = 0)
row.names(merged.df) <- merged.df$Row.names
merged.df <- merged.df[,-1]
```

```{r Determining the specificity score to use as a cutoff}
color <- c("#F2C318","#E790AC","#F18421","#0168A5","#848483","#A4CAEB","#885793","#008957","#222222")
color.transparent <- adjustcolor(color, alpha.f = 0.3)
bins <- seq(from = 0, to = 1, by= 0.005)

hist(dat.JSD$P7.FB.1.mean, breaks = bins, col = color.transparent[1], main = "P7 Subset Cluster Specificity Distributions", ylab = "Genes", xlab = "Specificity score")
hist(dat.JSD$P7.FB.2.mean, breaks = bins, add = T, col = color.transparent[2])
hist(dat.JSD$P7.MB.1.mean, breaks = bins, add = T, col = color.transparent[3])
hist(dat.JSD$P7.MB.2.mean, breaks = bins, add = T, col = color.transparent[4])
hist(dat.JSD$P7.MB.3.mean, breaks = bins, add = T, col = color.transparent[5])
hist(dat.JSD$P7.MB.4.mean, breaks = bins, add = T, col = color.transparent[6])
hist(dat.JSD$P7.OB.1.mean, breaks = bins, add = T, col = color.transparent[7])
hist(dat.JSD$P7.OB.2.mean, breaks = bins, add = T, col = color.transparent[8])
hist(dat.JSD$P7.OB.3.mean, breaks = bins, add = T, col = color.transparent[9])
abline(v = 0.3, lwd = 2)
```

```{r determining the specific genes}
P7.specific.genes <- data.frame()

for(i in 1:9){
  gene.tmp.1 <- merged.df[merged.df[,i] >= 0.3 & merged.df[,i+9] >= 0.4,]
  gene.ordered <- row.names(gene.tmp.1[order(-gene.tmp.1[,i]),])
  cluster.genes <- data.frame(lookupGeneName(P7.dat.filter, gene.ordered))
  P7.specific.genes <- cbindPad(P7.specific.genes, cluster.genes)
}

names(P7.specific.genes) <- c("P7.FB.1","P7.FB.2","P7.MB.1","P7.MB.2","P7.MB.3","P7.MB.4","P7.OB.1","P7.OB.2","P7.OB.3")

write.table(P7.specific.genes, file = "/Volumes/PAULHOOK/Revision/P7.specific.genes.txt", sep = "\t", quote = F, row.names = F)
```

```{r}
dat.filter <- readRDS(file = file.path(Rdata_dir, "dat.filter.final.Rds"))
nrow(pData(dat.filter)) # 396
dat.levels <- levels(pData(dat.filter)$subset.cluster)
cluster.mean.df <- as.data.frame(rownames(exprs(dat.filter)))
colnames(cluster.mean.df) <- "gene"

for(i in 1:length(dat.levels)){
  tmp <- dat.filter[,pData(dat.filter)$subset.cluster == dat.levels[i]]
  tmp.exprs <- as.matrix(exprs(tmp))
  tmp.mean <- as.data.frame(apply(tmp.exprs, 1, mean))
  row.names(tmp.mean) <- row.names(tmp.exprs)
  colnames(tmp.mean) <- paste0(dat.levels[i],".Mean")
  cluster.mean.df <- merge(x = cluster.mean.df, y = tmp.mean, by.x = "gene", by.y = 0)
}

rownames(cluster.mean.df) <- cluster.mean.df$gene
cluster.mean.df <- cluster.mean.df[,-1]

gwas.genes <- read.delim(file = "/Volumes/PAULHOOK/Revision/PD.GWAS.Prioritization.Revise/PD.GWAS.Prioritization.RevisePD.GWAS.Score.Final.txt")

gwas.gene.exprs <- gwas.genes[gwas.genes$lin.exprs == 2 & gwas.genes$expression == 2,]
gwas.genes <- lookupGeneId(dat.filter, gene_names = gwas.gene.exprs$MouseSymbol)
gwas.cluster.mean <- cluster.mean.df[rownames(cluster.mean.df) %in% gwas.genes,]
gwas.cluster.mean.t <- t(gwas.cluster.mean)

test <- kmeans(gwas.cluster.mean, centers = 10, iter.max = 1000)
t.centers <- t(test$centers)
apply(t.centers, 2, plot)

tmp <- as.data.frame(cor(t(cluster.mean.df), t.centers[,1], method = "pearson"))
for(i in 2:ncol(t.centers)){
  test.1 <- as.data.frame(cor(t(cluster.mean.df), t.centers[,i], method = "pearson"))
  tmp[,i] <- test.1[,1]
}
cor.df <- tmp
names(cor.df) <- c("k.1","k.2","k.3","k.4","k.5")
k.1.top100 <- rownames(cor.df[order(-cor.df$k.5),])[1:100]
k.5.names <- lookupGeneName(dat.filter, k.5.top100)

myBoxplot.subset(dat.filter, k.5.names[1:10])
k.5.names[k.5.names %in% gwas.gene.exprs$MouseSymbol]
plot(t.centers[,5])
```

```{r Performing GO Analyses on SN markers}
library(clusterProfiler)
markers.df <- read.delim(file = "/Volumes/PAULHOOK/Revision/P7.specific.genes.txt")

sn.genes <- as.character(markers.df$P7.MB.4)

result <- enrichGO(sn.genes,
                   ont = "BP",
                   OrgDb = "org.Mm.eg.db", 
                   keytype = "SYMBOL",
                   qvalueCutoff = 0.01)

result.df <- summary(result)
```
