---
title: "P7 MB Recursive Analysis"
author: "Paul Hook"
output: html_document
editor_options: 
  chunk_output_type: console
---

**Last update:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

###Setting important directories
Also loading important libraries and custom functions for analysis.
```{r init, message=FALSE, warning=FALSE}
seq_dir <- "/Volumes/PAULHOOK/sc-da-parkinsons/data"
file_dir <- "/Volumes/PAULHOOK/sc-da-parkinsons/output"
Rdata_dir <- "/Volumes/PAULHOOK/sc-da-parkinsons/data"
Script_dir <- "/Volumes/PAULHOOK/sc-da-parkinsons/code"
source(file.path(Script_dir,'init.R'))
source(file.path(Script_dir,"tools_R.r"))
```

###Loading MB .rds
```{r loading .rds}
P7.Mb.dat.filter <- readRDS(file = file.path(Rdata_dir, "P7.Mb.dat.filter.rds"))
```

###Filter genes by percentage of cells expresssing each gene
```{r filter by cells expressing each gene}
# Plot number of cells expressing each gene as histogram
hist(fData(P7.Mb.dat.filter)$num_cells_expressed,breaks=100,col="red",main="Cells expressed per gene")

# Keep only expressed genes with expression in >= 5% of cells
numCellThreshold<-nrow(pData(P7.Mb.dat.filter))*0.05
P7.Mb.dat.expressed_genes<-row.names(subset(fData(P7.Mb.dat.filter),num_cells_expressed >= numCellThreshold))

# Same plot as above with threshold
hist(fData(P7.Mb.dat.filter)$num_cells_expressed,breaks=100,col="red",main="Cells expressed per gene - threshold")
abline(v=numCellThreshold,lty="dashed")
```

###Prepping the Monocle model for analysis
```{r model_prep - Preparing the data for monocle analysis}
# Only keeping "expressed" genes
P7.Mb.dat.filter <-P7.Mb.dat.filter[P7.Mb.dat.expressed_genes,]

# Estimating the size factors
P7.Mb.dat.filter <-estimateSizeFactors(P7.Mb.dat.filter)

# Estimating dispersions
P7.Mb.dat.filter <- estimateDispersions(P7.Mb.dat.filter,cores=8)
# Removing 144 outliers
# Warning message:
# Deprecated, use tibble::rownames_to_column() instead. 
```

###Calculating summary stats
```{r summary_stats}
# Calculating summary stats
fData(P7.Mb.dat.filter)$mean_expr<-apply(round(exprs(P7.Mb.dat.filter)),1,mean) # mean expression
fData(P7.Mb.dat.filter)$sd_expr<-apply(round(exprs(P7.Mb.dat.filter)),1,sd) # sd expression
fData(P7.Mb.dat.filter)$bcv<-(fData(P7.Mb.dat.filter)$sd_expr/fData(P7.Mb.dat.filter)$mean_expr)**2 # calculating biological coefficient of variation
fData(P7.Mb.dat.filter)$percent_detection<-(fData(P7.Mb.dat.filter)$num_cells_expressed/dim(P7.Mb.dat.filter)[2])*100 # calculating % detection
```

###Identifying high dispersion genes
```{r high_dispersion_genes_monocle - Pulling out the high dispersion genes for PCA analysis}
P7.Mb.dat.filter.genes <- P7.Mb.dat.filter # spoofing the CellDataSet
disp_table <- dispersionTable(P7.Mb.dat.filter.genes) # pulling out the dispersion table
unsup_clustering_genes <-subset(disp_table, mean_expression >= 0.5 & dispersion_empirical >= 1.5 * dispersion_fit) # subsetting the data to pull out genes with expression above 0.5 and dispersion empirical > 2
P7.Mb.dat.high_bcv_genes<-unsup_clustering_genes$gene_id # pulling out list of genes
P7.Mb.dat.filter.order <- setOrderingFilter(P7.Mb.dat.filter, unsup_clustering_genes$gene_id)
plot_ordering_genes(P7.Mb.dat.filter.order) # plotting the dispersion and genes
length(P7.Mb.dat.high_bcv_genes) # 923
```

###Running PCA with high dispersion genes
```{r Run PCA with high BCV genes}
# BCV Identified high dispersion genes. Running PC analysis
P7.Mb.dat.filter.BCV.pca<-prcomp(t(log2(exprs(P7.Mb.dat.filter[P7.Mb.dat.high_bcv_genes,])+1)),center=T,scale. = TRUE)

# Plotting the PCA graphs
# Plotting the first 2 PCs and coloring by age
hvPCA1<-ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$age,ellipse=T,var.axes=F) + scale_color_manual(values=c("darkgreen","red")) + monocle:::monocle_theme_opts()

# Plotting the first 2 PCs and coloring by region
hvPCA2<-ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$region,ellipse=T,var.axes=F) + scale_color_brewer(palette="Set1") + monocle:::monocle_theme_opts() + ggtitle("P7 MB PCA")

# Plotting the first 2 PCs and coloring by plate the cell was sequenced from
hvPCA3<-ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$split_plate,ellipse=T, var.axes=F) + scale_color_brewer(palette="Set1") + monocle:::monocle_theme_opts()

# Show the plots in the terminal
hvPCA1
hvPCA2
hvPCA3
```

###Outlier identification
The first pass analysis seems to show obvious outliers in the PCA. Below, the relevant data is saved to look at outliers in the future
```{r saving .rds of outliers}
MB.outlier.plot <- ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$region,ellipse=F,var.axes=F)

saveRDS(MB.outlier.plot, file = file.path(Rdata_dir,"P7.MB.PCA.outliers.rds"))

saveRDS(hvPCA2, file = file.path(Rdata_dir,"P7.MB.PCA.outliers.rds"))
```

###Outlier analysis
In an attempt to identify what the outliers are, genes driving the outlier appearance are extracted and saved for GSEA analysis
```{r what are these outliers}
# PC1 seems to describe outliers in the data
P7.MB.dat.pca.rotations.df <- as.data.frame(P7.Mb.dat.filter.BCV.pca$rotation[,1:6])
genes.tmp <- P7.MB.dat.pca.rotations.df[order(-P7.MB.dat.pca.rotations.df$PC1),]
PC1.genes <- as.character(row.names(genes.tmp))
PC1.genes.names <- lookupGeneName(P7.Mb.dat.filter,PC1.genes)
genes.tmp$gene_short_name <- PC1.genes.names

PC1.genes.df <- genes.tmp[,c(7,1)]
write.table(PC1.genes.df, file = file.path(file_dir, "P7.Mb.cells.PC1.rnk"),quote = F, sep = '\t', row.names = F, col.names = F)

# PC2
P7.MB.dat.pca.rotations.df <- as.data.frame(P7.Mb.dat.filter.BCV.pca$rotation[,1:6])
genes.tmp <- P7.MB.dat.pca.rotations.df[order(-P7.MB.dat.pca.rotations.df$PC2),]
PC2.genes <- as.character(row.names(genes.tmp))
PC2.genes.names <- lookupGeneName(P7.Mb.dat.filter,PC2.genes)
genes.tmp$gene_short_name <- PC2.genes.names

PC2.genes.df <- genes.tmp[,c(7,2)]
write.table(PC2.genes.df, file = file.path(file_dir, "P7.Mb.cells.PC2.rnk"),quote = F, sep = '\t', row.names = F, col.names = F)
```

###Removing outliers from the original MB dataset
```{r removing outliers}
pData(P7.Mb.dat.filter)$PC1 <- P7.Mb.dat.filter.BCV.pca$x[,1]
pData(P7.Mb.dat.filter)$PC2 <- P7.Mb.dat.filter.BCV.pca$x[,2]
P7.Mb.dat.outliers.1<-as.character(pData(P7.Mb.dat.filter[,pData(P7.Mb.dat.filter)$PC2 >= 15])$sample_id.x)
P7.Mb.dat.outliers.2<-as.character(pData(P7.Mb.dat.filter[,pData(P7.Mb.dat.filter)$PC1 >= 15])$sample_id.x)
P7.Mb.outliers <- c(P7.Mb.dat.outliers.1, P7.Mb.dat.outliers.2)
P7.Mb.outliers <- unique(P7.Mb.outliers)
length(P7.Mb.outliers)

P7.Mb.dat.filter <- readRDS(file = file.path(Rdata_dir, "P7.Mb.dat.filter.rds"))

P7.Mb.dat.filter <- P7.Mb.dat.filter[,!(pData(P7.Mb.dat.filter)$sample_id.x %in% P7.Mb.outliers)]

nrow(pData(P7.Mb.dat.filter)) #75

saveRDS(object = P7.Mb.outliers, file = file.path(Rdata_dir,"P7.Mb.outliers.rds"))
```

###Rerunning analysis on P7 MB with outliers removed.
Starting here, all work performed above was re-run with the outliers removed.

###Filter genes by percentage of cells expresssing each gene with outliers removed
```{r filter by cells expressing each gene with outliers removed}
# Plot number of cells expressing each gene as histogram
hist(fData(P7.Mb.dat.filter)$num_cells_expressed,breaks=100,col="red",main="Cells expressed per gene")

# Keep only expressed genes with expression in >= 5% of cells
numCellThreshold<-nrow(pData(P7.Mb.dat.filter))*0.05
P7.Mb.dat.expressed_genes<-row.names(subset(fData(P7.Mb.dat.filter),num_cells_expressed >= numCellThreshold))

# Same plot as above with threshold
hist(fData(P7.Mb.dat.filter)$num_cells_expressed,breaks=100,col="red",main="Cells expressed per gene - threshold")
abline(v=numCellThreshold,lty="dashed")
```

###Prepping the Monocle model for analysis with outliers removed
```{r model_prep - Preparing the data for monocle analysis with outliers removed}
# Only keeping "expressed" genes
P7.Mb.dat.filter <-P7.Mb.dat.filter[P7.Mb.dat.expressed_genes,]

# Estimating the size factors
P7.Mb.dat.filter <-estimateSizeFactors(P7.Mb.dat.filter)

# Estimating dispersions
P7.Mb.dat.filter <- estimateDispersions(P7.Mb.dat.filter,cores=8)
# Removing 221 outliers
# Warning message:
# Deprecated, use tibble::rownames_to_column() instead. 
```

###Calculating summary stats with outliers removed
```{r summary_stats with outliers removed}
# Calculating summary stats
fData(P7.Mb.dat.filter)$mean_expr<-apply(round(exprs(P7.Mb.dat.filter)),1,mean) # mean expression
fData(P7.Mb.dat.filter)$sd_expr<-apply(round(exprs(P7.Mb.dat.filter)),1,sd) # sd expression
fData(P7.Mb.dat.filter)$bcv<-(fData(P7.Mb.dat.filter)$sd_expr/fData(P7.Mb.dat.filter)$mean_expr)**2 # calculating biological coefficient of variation
fData(P7.Mb.dat.filter)$percent_detection<-(fData(P7.Mb.dat.filter)$num_cells_expressed/dim(P7.Mb.dat.filter)[2])*100 # calculating % detection
```

###Identifying high dispersion genes with outliers removed
```{r high_dispersion_genes_monocle - Pulling out the high dispersion genes for PCA analysis with outliers removed}
P7.Mb.dat.filter.genes <- P7.Mb.dat.filter # spoofing the CellDataSet
disp_table <- dispersionTable(P7.Mb.dat.filter.genes) # pulling out the dispersion table
unsup_clustering_genes <-subset(disp_table, mean_expression >= 0.5 & dispersion_empirical >= 1.5 * dispersion_fit) # subsetting the data to pull out genes with expression above 0.5 and dispersion empirical > 2
P7.Mb.dat.high_bcv_genes<-unsup_clustering_genes$gene_id # pulling out list of genes
P7.Mb.dat.filter.order <- setOrderingFilter(P7.Mb.dat.filter, unsup_clustering_genes$gene_id)
plot_ordering_genes(P7.Mb.dat.filter.order) # plotting the dispersion and genes
length(P7.Mb.dat.high_bcv_genes) # 1185
```

###Running PCA with high dispersion genes with outliers removed
```{r Run PCA with high BCV genes with outliers removed}
# BCV Identified high dispersion genes. Running PC analysis
P7.Mb.dat.filter.BCV.pca<-prcomp(t(log2(exprs(P7.Mb.dat.filter[P7.Mb.dat.high_bcv_genes,])+1)),center=T,scale. = TRUE)

# Plotting the PCA graphs
# Plotting the first 2 PCs and coloring by age
hvPCA1<-ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$age,ellipse=T,var.axes=F) + scale_color_manual(values=c("darkgreen","red")) + monocle:::monocle_theme_opts()

# Plotting the first 2 PCs and coloring by region
hvPCA2<-ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$region,ellipse=T,var.axes=F) + scale_color_brewer(palette="Set1") + monocle:::monocle_theme_opts()

# Plotting the first 2 PCs and coloring by plate the cell was sequenced from
hvPCA3<-ggbiplot(P7.Mb.dat.filter.BCV.pca,choices=c(1,2),scale=0,groups=pData(P7.Mb.dat.filter)$split_plate,ellipse=T,var.axes=F) + scale_color_brewer(palette="Set1") + monocle:::monocle_theme_opts()

# Show the plots in the terminal
hvPCA1
hvPCA2
hvPCA3

# No real outliers seen
```

###Screeplots
Viewing screeplots and determining the number of "significant" PCs. Since no additional outliers were identified in the PCA plot above, we will continue with the analysis with only the original outliers removed
```{r Screeplots of Data,tidy=TRUE}
# Making a screeplot of the BCV PCA. This will help determine how 
# many principal components we should use in our tSNE visualization
# Show this plot
screeplot(P7.Mb.dat.filter.BCV.pca, npcs = 30, main = "P7 MB - High BCV Genes PCA Screeplot")
abline(v=5.5,lwd=2,col="red")
ggscreeplot(P7.Mb.dat.filter.BCV.pca, type = "pev")

# Conclustion: Seems to be clear that just the 
# first 5 PCs explain the most variation in our data
```

###Creating a t-SNE plot from the "significant" PCs with outliers removed
```{r tsne with outliers removed, message=FALSE,warning=FALSE}
nComponents<-5 # estimated from the screeplots
#seed <- runif(1,1,9999) # determined by testing random seeds
seed <- 7641.869
set.seed(seed) #setting seed

P7.Mb.dat.filter.BCV.tsne<-tsne(P7.Mb.dat.filter.BCV.pca$x[,1:nComponents],perplexity=10,max_iter=5000,whiten = FALSE)

pData(P7.Mb.dat.filter)$tSNE1_pos<-P7.Mb.dat.filter.BCV.tsne[,1]
pData(P7.Mb.dat.filter)$tSNE2_pos<-P7.Mb.dat.filter.BCV.tsne[,2]

P7.Mb.dat.filter.BCV.tsne.plot<-myTSNEPlotAlpha(P7.Mb.dat.filter,color="region", shape="age") + scale_color_brewer(palette="Set1") + ggtitle("P7 Mb - BCV tSNE Plot")

P7.Mb.dat.filter.BCV.tsne.plot
```

###Identifying clusters
Identifying clusters in the data in an unsupervised manner with outliers removed
```{r - attempting to cluster,tidy=TRUE}
# Going to attempt to use the R program "ADPclust" to determine how many clusters our data has

# Loading NbClust
library(ADPclust)

# Running ADPclust
clust.res <- adpclust(x = P7.Mb.dat.filter.BCV.tsne)

# Extracting the "best partition" (aka the best cluster) for each cell
clust.res.df <- as.data.frame(clust.res$cluster)

# Adding the cluster assignment for each cell to the pData
pData(P7.Mb.dat.filter)$kmeans_tSNE_cluster <- as.factor(clust.res.df$`clust.res$cluster`)

# Plotting the same tSNE plot as above but coloring with the "clusters"
myTSNEPlotAlpha(P7.Mb.dat.filter,color="kmeans_tSNE_cluster", shape="age") + scale_color_brewer(palette="Set1") + ggtitle("P7 Mb - BCV tSNE with Clusters Plot")
```

###QC on the clusters with outliers removed
```{r check the clusters - qc on clusters with outliers removed}
# Distribution of number of genes expressed in clusters
q <- ggplot(pData(P7.Mb.dat.filter)) +
  geom_density(aes(x=num_genes_expressed,fill=kmeans_tSNE_cluster),alpha=0.3) + scale_fill_brewer(palette="Set1") + facet_grid(.~age) + monocle:::monocle_theme_opts()

q

# Plotting the distribution of total mRNAs in clusters
q<-ggplot(pData(P7.Mb.dat.filter)) +
  geom_density(aes(x=Total_mRNAs,fill=kmeans_tSNE_cluster),alpha=0.3) + scale_fill_brewer(palette="Set1") + facet_grid(.~age) + monocle:::monocle_theme_opts()

q
```

###Saving the P7 MB final cds 
```{r saving analyzed cds for P7 Mb}
P7.Mb.dat.filter.final <- P7.Mb.dat.filter
saveRDS(object = P7.Mb.dat.filter.final, file = file.path(Rdata_dir, "P7.Mb.dat.filter.final.Rds"))
```

###Extracting pData information and saving it 
```{r extacting pData information}
P7.Mb.clusters.df <- pData(P7.Mb.dat.filter)[,c(1,39)]
saveRDS(P7.Mb.clusters.df, file = file.path(Rdata_dir, "P7.Mb.clusters.df.rds"))
```

###Extracting SN expressed genes
In order to score the PD GWAS genes later, we determined the genes expressed in SN DA neurons
```{r SN expressed}
# Extracting just the SN DA neurons
sn_exprs <- P7.Mb.dat.filter[,pData(P7.Mb.dat.filter)$kmeans_tSNE_cluster == 4]

# Finding the average expression for each gene
sn.average <- as.data.frame(apply(exprs(sn_exprs),1, function(x) mean(x)))
names(sn.average) <- "average"

# Adding a column so dataframe functions can be used
sn.average$average.2 <- sn.average$average

# Only keeping genes that have an an average >= 0.5 transcripts
sn.expressed <- sn.average[which(sn.average$average >= 0.5),]

# Looking up gene names
sn.expressed$gene <- lookupGeneName(P7.Mb.dat.filter, row.names(sn.expressed))

#Counting number of rowns
nrow(sn.expressed)

#Kepping only unique gene names
SN.expressed.genes <- unique(sn.expressed$gene)

#Saving the rds for later scoring
saveRDS(SN.expressed.genes, file = file.path(Rdata_dir,"SN.expressed.genes.rds"))
```

###Session Info
```{r session infomation with outliers removed}
sessionInfo()
```
